In [1]:
import uuid
from functools import partial

import evaluate
import numpy as np
from datasets import load_dataset
from transformers import (
    AutoModelForTokenClassification,
    AutoTokenizer,
    DataCollatorForTokenClassification,
    Trainer,
    TrainingArguments,
)


def is_subword(text, tokenized, tokenizer, index):
    word = tokenizer.convert_ids_to_tokens(tokenized["input_ids"][index])
    start_ind, end_ind = tokenized["offset_mapping"][index]
    word_ref = text[start_ind:end_ind]
    is_subword = len(word) != len(word_ref)
    return is_subword


def tokenize(example, labels2int, tokenizer, iob=True, ignore_subwords=True):

    text, labels = example["source_text"], example["privacy_mask"]

    i = 0
    token_labels = []

    tokenized = tokenizer(text, return_offsets_mapping=True, return_special_tokens_mask=True)
    start_token_to_label = {
        tokenized.char_to_token(label["start"]): (label["start"], label["end"], label["label"]) for label in labels
    }
    while i < len(tokenized["input_ids"]):
        if tokenized["special_tokens_mask"][i] == 1:
            token_labels.append(-100)
            i += 1
        elif i not in start_token_to_label:
            if ignore_subwords and is_subword(text, tokenized, tokenizer, i):
                token_labels.append(-100)
            else:
                token_labels.append(labels2int["O"])
            i += 1
        else:
            start, end, label = start_token_to_label[i]
            start_token = tokenized.char_to_token(start)
            assert start_token == i
            j = start_token
            while j < (len(tokenized["input_ids"]) - 1) and tokenized.token_to_chars(j).start < end:
                if j == start_token:
                    if iob:
                        token_labels.append(labels2int["B-" + label])
                    else:
                        token_labels.append(labels2int[label])
                elif ignore_subwords and is_subword(text, tokenized, tokenizer, j):
                    token_labels.append(-100)
                else:
                    if iob:
                        token_labels.append(labels2int["I-" + label])
                    else:
                        token_labels.append(labels2int[label])

                j += 1
            i = j
    tokenized["labels"] = token_labels
    return tokenized


def compute_metrics(eval_pred, label_list, seqeval_metric):
    predictions, labels = eval_pred
    predictions = np.argmax(predictions, axis=2)

    # Remove ignored index (special tokens)
    true_predictions = [
        [label_list[p] for (p, l) in zip(prediction, label) if l != -100]
        for prediction, label in zip(predictions, labels)
    ]
    true_labels = [
        [label_list[l] for (p, l) in zip(prediction, label) if l != -100]
        for prediction, label in zip(predictions, labels)
    ]

    results = seqeval_metric.compute(predictions=true_predictions, references=true_labels)
    results_flat = {f"{k}_f1": v["f1"] for k, v in results.items() if isinstance(v, dict)}
    results_flat.update(
        {
            "precision": results["overall_precision"],
            "recall": results["overall_recall"],
            "f1": results["overall_f1"],
            "accuracy": results["overall_accuracy"],
        }
    )
    return results_flat


/usr/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
2024-09-21 13:57:27.829353: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-09-21 13:57:27.913210: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-09-21 13:57:27.926945: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-09-21 13:57:28.019258: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use av

In [2]:
labels = [
    "BOD",
    "BUILDING",
    "CARDISSUER",
    "CITY",
    "COUNTRY",
    "DATE",
    "DRIVERLICENSE",
    "EMAIL",
    "GEOCOORD",
    "GIVENNAME1",
    "GIVENNAME2",
    "IDCARD",
    "IP",
    "LASTNAME1",
    "LASTNAME2",
    "LASTNAME3",
    "PASS",
    "PASSPORT",
    "POSTCODE",
    "SECADDRESS",
    "SEX",
    "SOCIALNUMBER",
    "STATE",
    "STREET",
    "TEL",
    "TIME",
    "TITLE",
    "USERNAME",
]

labels = [f"I-{label}" for label in labels] + [f"B-{label}" for label in labels] + ["O"]
label2id = {label: i for i, label in enumerate(labels)}
id2label = {v: k for k, v in label2id.items()}
pretrained_name = "distilbert-base-cased"
tokenizer = AutoTokenizer.from_pretrained(pretrained_name)
model = AutoModelForTokenClassification.from_pretrained(pretrained_name, num_labels=len(labels), id2label=id2label)

ds = load_dataset("ai4privacy/pii-masking-300k")
ds = ds.filter(lambda x: x["language"] == "English", num_proc=4)
ds = ds.map(
    partial(tokenize, labels2int=label2id, tokenizer=tokenizer, iob=True, ignore_subwords=True),
    batched=False,
    remove_columns=[
        "source_text",
        "target_text",
        "privacy_mask",
        "span_labels",
        "mbert_text_tokens",
        "mbert_bio_labels",
        "id",
        "language",
        "set",
    ],
    num_proc=8,
).remove_columns(["offset_mapping"])

training_arguments = TrainingArguments(
    output_dir="output",
    num_train_epochs=50,
    per_device_train_batch_size=32,
    per_device_eval_batch_size=32,
    overwrite_output_dir=True,
    warmup_ratio=0.2,
    learning_rate=2e-5,
    weight_decay=0.01,
    save_strategy="epoch",
    eval_strategy="epoch",
    load_best_model_at_end=True,
    lr_scheduler_type="cosine",
    warmup_steps=3000,
    metric_for_best_model="f1",
    greater_is_better=True,
    torch_compile=False,
)
trainer = Trainer(
    model,
    training_arguments,
    train_dataset=ds["train"],
    eval_dataset=ds["validation"],
    data_collator=DataCollatorForTokenClassification(tokenizer),
    tokenizer=tokenizer,
    compute_metrics=partial(compute_metrics, label_list=labels, seqeval_metric=evaluate.load("seqeval")),
)


/home/arnesh/.local/lib/python3.12/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(
Some weights of DistilBertForTokenClassification were not initialized from the model checkpoint at distilbert-base-cased and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Using the latest cached version of the dataset since ai4privacy/pii-masking-300k couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/arnesh/.cache/huggingface/datasets/ai4privacy___pii-masking-300k/default/0.0.0/3b99ae5980049a0dff660f424916afa995

In [3]:
trainer.train()


  1%|          | 500/46750 [02:25<3:31:26,  3.65it/s]

{'loss': 1.8906, 'grad_norm': 1.5857051610946655, 'learning_rate': 3.3333333333333333e-06, 'epoch': 0.53}


  2%|▏         | 935/46750 [04:30<3:11:14,  3.99it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                     
  2%|▏         | 935/46750 [05:04<3:11:14,  3.99it/s]

{'eval_loss': 0.15687373280525208, 'eval_BOD_f1': 0.8350714561606798, 'eval_BUILDING_f1': 0.8203125, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.475298126064736, 'eval_COUNTRY_f1': 0.7111431662459431, 'eval_DATE_f1': 0.674775557486244, 'eval_DRIVERLICENSE_f1': 0.5127231295100645, 'eval_EMAIL_f1': 0.8675543097407148, 'eval_GEOCOORD_f1': 0.2532188841201717, 'eval_GIVENNAME1_f1': 0.42192853646598144, 'eval_GIVENNAME2_f1': 0.0, 'eval_IDCARD_f1': 0.3282051282051282, 'eval_IP_f1': 0.8661864221785637, 'eval_LASTNAME1_f1': 0.4767580452920143, 'eval_LASTNAME2_f1': 0.0, 'eval_LASTNAME3_f1': 0.0, 'eval_PASS_f1': 0.5112866817155756, 'eval_PASSPORT_f1': 0.5995800839832034, 'eval_POSTCODE_f1': 0.626850168787328, 'eval_SECADDRESS_f1': 0.14598540145985403, 'eval_SEX_f1': 0.7146829810901001, 'eval_SOCIALNUMBER_f1': 0.5713782164258018, 'eval_STATE_f1': 0.7562836624775583, 'eval_STREET_f1': 0.4586220953933958, 'eval_TEL_f1': 0.6178059666431759, 'eval_TIME_f1': 0.8856878551605656, 'eval_TITLE_f1': 0.5777

  2%|▏         | 1000/46750 [05:24<3:43:52,  3.41it/s] 

{'loss': 0.2844, 'grad_norm': 1.108330488204956, 'learning_rate': 6.666666666666667e-06, 'epoch': 1.07}


  3%|▎         | 1500/46750 [07:49<3:45:12,  3.35it/s]

{'loss': 0.1285, 'grad_norm': 1.0395864248275757, 'learning_rate': 1e-05, 'epoch': 1.6}


  4%|▍         | 1870/46750 [09:37<3:16:44,  3.80it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
  4%|▍         | 1870/46750 [10:09<3:16:44,  3.80it/s]

{'eval_loss': 0.06610789149999619, 'eval_BOD_f1': 0.9261772568373433, 'eval_BUILDING_f1': 0.9602430995188654, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.8779482482253264, 'eval_COUNTRY_f1': 0.9348101265822785, 'eval_DATE_f1': 0.8484136310223267, 'eval_DRIVERLICENSE_f1': 0.8279895140149224, 'eval_EMAIL_f1': 0.9721909056745585, 'eval_GEOCOORD_f1': 0.9339407744874715, 'eval_GIVENNAME1_f1': 0.6681931723563697, 'eval_GIVENNAME2_f1': 0.09256198347107437, 'eval_IDCARD_f1': 0.8298404153047491, 'eval_IP_f1': 0.9441786283891547, 'eval_LASTNAME1_f1': 0.5937172774869109, 'eval_LASTNAME2_f1': 0.0889487870619946, 'eval_LASTNAME3_f1': 0.0, 'eval_PASS_f1': 0.7784642541924095, 'eval_PASSPORT_f1': 0.8467238689547582, 'eval_POSTCODE_f1': 0.9356603282104714, 'eval_SECADDRESS_f1': 0.9211428571428572, 'eval_SEX_f1': 0.9386963896292706, 'eval_SOCIALNUMBER_f1': 0.8376383763837638, 'eval_STATE_f1': 0.9316109422492401, 'eval_STREET_f1': 0.881653746770026, 'eval_TEL_f1': 0.9200193892389724, 'eval_TIME_f1': 0.9

  4%|▍         | 2000/46750 [10:50<3:28:13,  3.58it/s]  

{'loss': 0.0862, 'grad_norm': 0.9649495482444763, 'learning_rate': 1.3333333333333333e-05, 'epoch': 2.14}


  5%|▌         | 2500/46750 [13:14<3:50:02,  3.21it/s]

{'loss': 0.0691, 'grad_norm': 0.5306574106216431, 'learning_rate': 1.6666666666666667e-05, 'epoch': 2.67}


  6%|▌         | 2805/46750 [14:42<3:04:42,  3.97it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
  6%|▌         | 2805/46750 [15:15<3:04:42,  3.97it/s]

{'eval_loss': 0.04948381334543228, 'eval_BOD_f1': 0.9475032010243278, 'eval_BUILDING_f1': 0.9759097898513582, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9261418853255587, 'eval_COUNTRY_f1': 0.9516728624535314, 'eval_DATE_f1': 0.8861696072851452, 'eval_DRIVERLICENSE_f1': 0.8876068376068376, 'eval_EMAIL_f1': 0.9790954009882175, 'eval_GEOCOORD_f1': 0.9515011547344111, 'eval_GIVENNAME1_f1': 0.74565560821485, 'eval_GIVENNAME2_f1': 0.4069431051108968, 'eval_IDCARD_f1': 0.8401796108431732, 'eval_IP_f1': 0.9762399077277971, 'eval_LASTNAME1_f1': 0.685282140779523, 'eval_LASTNAME2_f1': 0.3489813994685562, 'eval_LASTNAME3_f1': 0.0, 'eval_PASS_f1': 0.8436754176610978, 'eval_PASSPORT_f1': 0.8673311184939091, 'eval_POSTCODE_f1': 0.96248382923674, 'eval_SECADDRESS_f1': 0.9675174013921114, 'eval_SEX_f1': 0.9636184857423796, 'eval_SOCIALNUMBER_f1': 0.8945777061076396, 'eval_STATE_f1': 0.9594829729057918, 'eval_STREET_f1': 0.9249370277078085, 'eval_TEL_f1': 0.9356181150550795, 'eval_TIME_f1': 0.954943

  6%|▋         | 3000/46750 [16:13<3:47:03,  3.21it/s]  

{'loss': 0.0561, 'grad_norm': 0.5773890614509583, 'learning_rate': 2e-05, 'epoch': 3.21}


  7%|▋         | 3500/46750 [18:36<3:31:31,  3.41it/s]

{'loss': 0.0491, 'grad_norm': 0.6346962451934814, 'learning_rate': 1.9993555236433216e-05, 'epoch': 3.74}


  8%|▊         | 3740/46750 [19:47<2:57:59,  4.03it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
  8%|▊         | 3740/46750 [20:20<2:57:59,  4.03it/s]

{'eval_loss': 0.044348739087581635, 'eval_BOD_f1': 0.9469387755102041, 'eval_BUILDING_f1': 0.9762573398008679, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9373639661426844, 'eval_COUNTRY_f1': 0.9634990560100692, 'eval_DATE_f1': 0.8933102652825837, 'eval_DRIVERLICENSE_f1': 0.8933391380060948, 'eval_EMAIL_f1': 0.9804591159172832, 'eval_GEOCOORD_f1': 0.9699769053117783, 'eval_GIVENNAME1_f1': 0.7837419924895075, 'eval_GIVENNAME2_f1': 0.4580645161290323, 'eval_IDCARD_f1': 0.8880714661061481, 'eval_IP_f1': 0.9756992205410363, 'eval_LASTNAME1_f1': 0.7258873069981954, 'eval_LASTNAME2_f1': 0.44247787610619477, 'eval_LASTNAME3_f1': 0.028985507246376812, 'eval_PASS_f1': 0.876645240281604, 'eval_PASSPORT_f1': 0.9094600081201787, 'eval_POSTCODE_f1': 0.9624254988338947, 'eval_SECADDRESS_f1': 0.9736995908825248, 'eval_SEX_f1': 0.9677578144228401, 'eval_SOCIALNUMBER_f1': 0.906312826698834, 'eval_STATE_f1': 0.9662195727769498, 'eval_STREET_f1': 0.9523318732285493, 'eval_TEL_f1': 0.9380872079017103, 'e

  9%|▊         | 4000/46750 [21:37<3:42:58,  3.20it/s]  

{'loss': 0.0399, 'grad_norm': 0.24771149456501007, 'learning_rate': 1.9974229252728345e-05, 'epoch': 4.28}


 10%|▉         | 4500/46750 [24:03<3:20:23,  3.51it/s]

{'loss': 0.0374, 'grad_norm': 0.41228941082954407, 'learning_rate': 1.9942046959164516e-05, 'epoch': 4.81}


 10%|█         | 4675/46750 [24:53<2:50:04,  4.12it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
 10%|█         | 4675/46750 [25:26<2:50:04,  4.12it/s]

{'eval_loss': 0.04065316915512085, 'eval_BOD_f1': 0.9471184977336499, 'eval_BUILDING_f1': 0.9781209781209781, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9480137322216774, 'eval_COUNTRY_f1': 0.9691823899371069, 'eval_DATE_f1': 0.9054209919261822, 'eval_DRIVERLICENSE_f1': 0.9165239726027398, 'eval_EMAIL_f1': 0.9791587722622206, 'eval_GEOCOORD_f1': 0.928888888888889, 'eval_GIVENNAME1_f1': 0.8139759036144578, 'eval_GIVENNAME2_f1': 0.5938494167550371, 'eval_IDCARD_f1': 0.9181969949916527, 'eval_IP_f1': 0.9741122565864834, 'eval_LASTNAME1_f1': 0.7556368068251065, 'eval_LASTNAME2_f1': 0.5245441795231416, 'eval_LASTNAME3_f1': 0.1181818181818182, 'eval_PASS_f1': 0.8806835520292952, 'eval_PASSPORT_f1': 0.9226464121527084, 'eval_POSTCODE_f1': 0.9737532808398951, 'eval_SECADDRESS_f1': 0.9748096074985355, 'eval_SEX_f1': 0.9734557181840734, 'eval_SOCIALNUMBER_f1': 0.929024583663759, 'eval_STATE_f1': 0.9745635910224438, 'eval_STREET_f1': 0.9527458492975734, 'eval_TEL_f1': 0.9497536945812808, 'eval_

 11%|█         | 5000/46750 [27:01<3:23:08,  3.43it/s]  

{'loss': 0.0337, 'grad_norm': 0.34941795468330383, 'learning_rate': 1.989704983719635e-05, 'epoch': 5.35}


 12%|█▏        | 5500/46750 [29:26<3:11:09,  3.60it/s]

{'loss': 0.0315, 'grad_norm': 0.505343496799469, 'learning_rate': 1.98392958859863e-05, 'epoch': 5.88}


 12%|█▏        | 5610/46750 [29:58<2:54:09,  3.94it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
 12%|█▏        | 5610/46750 [30:31<2:54:09,  3.94it/s]

{'eval_loss': 0.042270295321941376, 'eval_BOD_f1': 0.9576070583171938, 'eval_BUILDING_f1': 0.9799485861182519, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9577187807276302, 'eval_COUNTRY_f1': 0.9653870358716173, 'eval_DATE_f1': 0.8843689049874616, 'eval_DRIVERLICENSE_f1': 0.8862973760932945, 'eval_EMAIL_f1': 0.9823696682464456, 'eval_GEOCOORD_f1': 0.9675925925925926, 'eval_GIVENNAME1_f1': 0.8221667045196457, 'eval_GIVENNAME2_f1': 0.6606170598911071, 'eval_IDCARD_f1': 0.9118416137467314, 'eval_IP_f1': 0.977194194885971, 'eval_LASTNAME1_f1': 0.7631018321261184, 'eval_LASTNAME2_f1': 0.546225614927905, 'eval_LASTNAME3_f1': 0.4252873563218391, 'eval_PASS_f1': 0.8856968215158925, 'eval_PASSPORT_f1': 0.9075777485764345, 'eval_POSTCODE_f1': 0.9709543568464729, 'eval_SECADDRESS_f1': 0.9772329246935202, 'eval_SEX_f1': 0.9730538922155688, 'eval_SOCIALNUMBER_f1': 0.9207304836443909, 'eval_STATE_f1': 0.9741422178020885, 'eval_STREET_f1': 0.9604753293722551, 'eval_TEL_f1': 0.9534883720930233, 'eval

 13%|█▎        | 6000/46750 [32:26<2:55:06,  3.88it/s]  

{'loss': 0.0248, 'grad_norm': 0.4960438013076782, 'learning_rate': 1.9768859547646476e-05, 'epoch': 6.42}


 14%|█▍        | 6500/46750 [34:51<3:05:17,  3.62it/s]

{'loss': 0.0253, 'grad_norm': 0.35618138313293457, 'learning_rate': 1.9685831611286312e-05, 'epoch': 6.95}


 14%|█▍        | 6545/46750 [35:03<2:49:04,  3.96it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
 14%|█▍        | 6545/46750 [35:36<2:49:04,  3.96it/s]

{'eval_loss': 0.044187162071466446, 'eval_BOD_f1': 0.9543604028283694, 'eval_BUILDING_f1': 0.9710108394252585, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.94564152791381, 'eval_COUNTRY_f1': 0.9580505860579889, 'eval_DATE_f1': 0.9009843659525187, 'eval_DRIVERLICENSE_f1': 0.9208163265306123, 'eval_EMAIL_f1': 0.982362981225109, 'eval_GEOCOORD_f1': 0.9744779582366588, 'eval_GIVENNAME1_f1': 0.8065023597273204, 'eval_GIVENNAME2_f1': 0.6624737945492662, 'eval_IDCARD_f1': 0.9254122938530734, 'eval_IP_f1': 0.978440366972477, 'eval_LASTNAME1_f1': 0.7690561755248723, 'eval_LASTNAME2_f1': 0.6041131105398458, 'eval_LASTNAME3_f1': 0.20689655172413793, 'eval_PASS_f1': 0.8869300911854103, 'eval_PASSPORT_f1': 0.933501153765471, 'eval_POSTCODE_f1': 0.9619537275064268, 'eval_SECADDRESS_f1': 0.971627099015634, 'eval_SEX_f1': 0.9650281242357543, 'eval_SOCIALNUMBER_f1': 0.9253616200578592, 'eval_STATE_f1': 0.9716399506781751, 'eval_STREET_f1': 0.946487446107025, 'eval_TEL_f1': 0.9547038327526132, 'eval_TIM

 15%|█▍        | 7000/46750 [37:49<2:56:39,  3.75it/s]  

{'loss': 0.0211, 'grad_norm': 0.2956547737121582, 'learning_rate': 1.959031909598966e-05, 'epoch': 7.49}


 16%|█▌        | 7480/46750 [40:09<2:55:31,  3.73it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
 16%|█▌        | 7480/46750 [40:42<2:55:31,  3.73it/s]

{'eval_loss': 0.04365336149930954, 'eval_BOD_f1': 0.9437609841827768, 'eval_BUILDING_f1': 0.9808526933877968, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9558823529411765, 'eval_COUNTRY_f1': 0.9631710362047441, 'eval_DATE_f1': 0.8939686036904434, 'eval_DRIVERLICENSE_f1': 0.9211843202668891, 'eval_EMAIL_f1': 0.9813546423135464, 'eval_GEOCOORD_f1': 0.9675925925925926, 'eval_GIVENNAME1_f1': 0.8369898560981364, 'eval_GIVENNAME2_f1': 0.6915725456125108, 'eval_IDCARD_f1': 0.9266873545384019, 'eval_IP_f1': 0.9770957398076042, 'eval_LASTNAME1_f1': 0.7831687419493345, 'eval_LASTNAME2_f1': 0.6102449888641426, 'eval_LASTNAME3_f1': 0.3882783882783883, 'eval_PASS_f1': 0.8968883465527762, 'eval_PASSPORT_f1': 0.9133216986620127, 'eval_POSTCODE_f1': 0.9766476388168137, 'eval_SECADDRESS_f1': 0.9795441262419637, 'eval_SEX_f1': 0.9715838893007166, 'eval_SOCIALNUMBER_f1': 0.929024583663759, 'eval_STATE_f1': 0.9759125900173826, 'eval_STREET_f1': 0.9545686574783052, 'eval_TEL_f1': 0.9539308963445168, 'eval

 16%|█▌        | 7500/46750 [40:49<3:22:47,  3.23it/s]  

{'loss': 0.0203, 'grad_norm': 1.1502939462661743, 'learning_rate': 1.9482445112872265e-05, 'epoch': 8.02}


 17%|█▋        | 8000/46750 [43:14<2:58:01,  3.63it/s]

{'loss': 0.017, 'grad_norm': 0.3819848299026489, 'learning_rate': 1.9362348706397374e-05, 'epoch': 8.56}


 18%|█▊        | 8415/46750 [45:14<3:03:57,  3.47it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
 18%|█▊        | 8415/46750 [45:48<3:03:57,  3.47it/s]

{'eval_loss': 0.04294256865978241, 'eval_BOD_f1': 0.9527288354103995, 'eval_BUILDING_f1': 0.9785056294779938, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9609240599655934, 'eval_COUNTRY_f1': 0.9696780243826195, 'eval_DATE_f1': 0.9109609168380841, 'eval_DRIVERLICENSE_f1': 0.9324867724867725, 'eval_EMAIL_f1': 0.9864632983794089, 'eval_GEOCOORD_f1': 0.960919540229885, 'eval_GIVENNAME1_f1': 0.8376068376068376, 'eval_GIVENNAME2_f1': 0.6976744186046512, 'eval_IDCARD_f1': 0.9246268656716418, 'eval_IP_f1': 0.9813062543272559, 'eval_LASTNAME1_f1': 0.7864693446088795, 'eval_LASTNAME2_f1': 0.6236722306525039, 'eval_LASTNAME3_f1': 0.445993031358885, 'eval_PASS_f1': 0.8958076448828607, 'eval_PASSPORT_f1': 0.9301860372074415, 'eval_POSTCODE_f1': 0.9753054328047829, 'eval_SECADDRESS_f1': 0.9748978400467017, 'eval_SEX_f1': 0.9751861042183623, 'eval_SOCIALNUMBER_f1': 0.940688018979834, 'eval_STATE_f1': 0.9776230730979611, 'eval_STREET_f1': 0.9653567735263702, 'eval_TEL_f1': 0.9620441577772265, 'eval_T

 18%|█▊        | 8500/46750 [46:13<3:26:57,  3.08it/s]  

{'loss': 0.0152, 'grad_norm': 0.18649035692214966, 'learning_rate': 1.9230184675153974e-05, 'epoch': 9.09}


 19%|█▉        | 9000/46750 [48:40<3:00:10,  3.49it/s]

{'loss': 0.014, 'grad_norm': 0.20294900238513947, 'learning_rate': 1.9086123372328748e-05, 'epoch': 9.63}


 20%|██        | 9350/46750 [50:21<3:01:06,  3.44it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                      
 20%|██        | 9350/46750 [50:54<3:01:06,  3.44it/s]

{'eval_loss': 0.045545462518930435, 'eval_BOD_f1': 0.9569500963803812, 'eval_BUILDING_f1': 0.9785823559408464, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9599413776257939, 'eval_COUNTRY_f1': 0.9679027734496728, 'eval_DATE_f1': 0.9163046626122212, 'eval_DRIVERLICENSE_f1': 0.9255784865540962, 'eval_EMAIL_f1': 0.9830766305381251, 'eval_GEOCOORD_f1': 0.9629629629629629, 'eval_GIVENNAME1_f1': 0.845984866975836, 'eval_GIVENNAME2_f1': 0.7146974063400576, 'eval_IDCARD_f1': 0.9206762028608583, 'eval_IP_f1': 0.9820193637621023, 'eval_LASTNAME1_f1': 0.798440065681445, 'eval_LASTNAME2_f1': 0.6577974870657798, 'eval_LASTNAME3_f1': 0.5813333333333333, 'eval_PASS_f1': 0.9034819792302994, 'eval_PASSPORT_f1': 0.9305954825462013, 'eval_POSTCODE_f1': 0.9766233766233767, 'eval_SECADDRESS_f1': 0.9767171129220024, 'eval_SEX_f1': 0.9752352649826647, 'eval_SOCIALNUMBER_f1': 0.9304959000390473, 'eval_STATE_f1': 0.9774473358116481, 'eval_STREET_f1': 0.9603705609881625, 'eval_TEL_f1': 0.9575848303393214, 'eval

 20%|██        | 9500/46750 [51:39<3:05:04,  3.35it/s]  

{'loss': 0.0127, 'grad_norm': 0.3332175314426422, 'learning_rate': 1.8930350486128855e-05, 'epoch': 10.16}


 21%|██▏       | 10000/46750 [54:04<2:45:34,  3.70it/s]

{'loss': 0.0121, 'grad_norm': 0.3113662302494049, 'learning_rate': 1.8763066800438638e-05, 'epoch': 10.7}


 22%|██▏       | 10285/46750 [55:26<2:36:51,  3.87it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                       
 22%|██▏       | 10285/46750 [55:59<2:36:51,  3.87it/s]

{'eval_loss': 0.044887855648994446, 'eval_BOD_f1': 0.9562216950614622, 'eval_BUILDING_f1': 0.9788103140158284, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9555446995603321, 'eval_COUNTRY_f1': 0.9676405906377631, 'eval_DATE_f1': 0.9005847953216375, 'eval_DRIVERLICENSE_f1': 0.9341093327790481, 'eval_EMAIL_f1': 0.9804219730089337, 'eval_GEOCOORD_f1': 0.9722222222222222, 'eval_GIVENNAME1_f1': 0.8422090729783037, 'eval_GIVENNAME2_f1': 0.718296224588577, 'eval_IDCARD_f1': 0.928153875165001, 'eval_IP_f1': 0.984295612009238, 'eval_LASTNAME1_f1': 0.7978853192354616, 'eval_LASTNAME2_f1': 0.6583072100313481, 'eval_LASTNAME3_f1': 0.5984251968503937, 'eval_PASS_f1': 0.9041769041769042, 'eval_PASSPORT_f1': 0.9373983739837399, 'eval_POSTCODE_f1': 0.9758253184299454, 'eval_SECADDRESS_f1': 0.9784004670169294, 'eval_SEX_f1': 0.9735867686990868, 'eval_SOCIALNUMBER_f1': 0.9429358520267612, 'eval_STATE_f1': 0.9775561097256857, 'eval_STREET_f1': 0.9638802889576884, 'eval_TEL_f1': 0.9589108910891089, 'eval_

 22%|██▏       | 10500/46750 [57:03<2:46:19,  3.63it/s]  

{'loss': 0.0105, 'grad_norm': 0.0853949710726738, 'learning_rate': 1.8584487936018663e-05, 'epoch': 11.23}


 24%|██▎       | 11000/46750 [59:28<2:35:33,  3.83it/s]

{'loss': 0.0092, 'grad_norm': 0.2126826047897339, 'learning_rate': 1.8394844072580772e-05, 'epoch': 11.76}


 24%|██▍       | 11220/46750 [1:00:32<2:26:00,  4.06it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                         
 24%|██▍       | 11220/46750 [1:01:05<2:26:00,  4.06it/s]

{'eval_loss': 0.047881633043289185, 'eval_BOD_f1': 0.9603789836347977, 'eval_BUILDING_f1': 0.9818460751725901, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9626168224299065, 'eval_COUNTRY_f1': 0.9695064445143037, 'eval_DATE_f1': 0.9105309474919331, 'eval_DRIVERLICENSE_f1': 0.933870284018652, 'eval_EMAIL_f1': 0.9830766305381251, 'eval_GEOCOORD_f1': 0.963302752293578, 'eval_GIVENNAME1_f1': 0.8321067157313707, 'eval_GIVENNAME2_f1': 0.7026143790849674, 'eval_IDCARD_f1': 0.9266089574708318, 'eval_IP_f1': 0.9852330410706046, 'eval_LASTNAME1_f1': 0.7800171526586622, 'eval_LASTNAME2_f1': 0.6490939044481054, 'eval_LASTNAME3_f1': 0.6029850746268657, 'eval_PASS_f1': 0.9000609384521633, 'eval_PASSPORT_f1': 0.9328015952143568, 'eval_POSTCODE_f1': 0.9768650896802703, 'eval_SECADDRESS_f1': 0.9784758580570099, 'eval_SEX_f1': 0.9740163325909429, 'eval_SOCIALNUMBER_f1': 0.941945710322964, 'eval_STATE_f1': 0.9751737835153923, 'eval_STREET_f1': 0.9658914728682171, 'eval_TEL_f1': 0.9653970624844411, 'eval_

 25%|██▍       | 11500/46750 [1:02:28<2:45:02,  3.56it/s]  

{'loss': 0.0075, 'grad_norm': 0.1442120373249054, 'learning_rate': 1.819437965209732e-05, 'epoch': 12.3}


 26%|██▌       | 12000/46750 [1:04:53<2:28:33,  3.90it/s]

{'loss': 0.008, 'grad_norm': 0.08684003353118896, 'learning_rate': 1.7983353063727014e-05, 'epoch': 12.83}


                                                         
 26%|██▌       | 12155/46750 [1:06:11<2:24:04,  4.00it/s]

{'eval_loss': 0.049103423953056335, 'eval_BOD_f1': 0.9559802447927851, 'eval_BUILDING_f1': 0.9808233188442853, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9612174766813941, 'eval_COUNTRY_f1': 0.9725292074518472, 'eval_DATE_f1': 0.9098384728340676, 'eval_DRIVERLICENSE_f1': 0.9243123870708692, 'eval_EMAIL_f1': 0.984434320425209, 'eval_GEOCOORD_f1': 0.9699769053117783, 'eval_GIVENNAME1_f1': 0.8432586946164841, 'eval_GIVENNAME2_f1': 0.7210884353741497, 'eval_IDCARD_f1': 0.9333588663347376, 'eval_IP_f1': 0.9796860572483841, 'eval_LASTNAME1_f1': 0.7918350848827809, 'eval_LASTNAME2_f1': 0.6502690238278247, 'eval_LASTNAME3_f1': 0.56957928802589, 'eval_PASS_f1': 0.8948012232415902, 'eval_PASSPORT_f1': 0.9335816263190565, 'eval_POSTCODE_f1': 0.9783119937287693, 'eval_SECADDRESS_f1': 0.9796393251890635, 'eval_SEX_f1': 0.9738142292490118, 'eval_SOCIALNUMBER_f1': 0.9409433214427269, 'eval_STATE_f1': 0.9770114942528736, 'eval_STREET_f1': 0.9648578811369508, 'eval_TEL_f1': 0.9610069101678184, 'eval_

 27%|██▋       | 12500/46750 [1:07:53<2:37:17,  3.63it/s]  

{'loss': 0.007, 'grad_norm': 0.29345929622650146, 'learning_rate': 1.7762036310763533e-05, 'epoch': 13.37}


 28%|██▊       | 13000/46750 [1:10:17<2:37:39,  3.57it/s]

{'loss': 0.007, 'grad_norm': 0.22719834744930267, 'learning_rate': 1.7530714660036112e-05, 'epoch': 13.9}


 28%|██▊       | 13090/46750 [1:10:43<2:40:40,  3.49it/s]/home/arnesh/.local/lib/python3.12/site-packages/seqeval/metrics/v1.py:57: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
                                                         
 28%|██▊       | 13090/46750 [1:11:16<2:40:40,  3.49it/s]

{'eval_loss': 0.051334675401449203, 'eval_BOD_f1': 0.9609057893612475, 'eval_BUILDING_f1': 0.9822758797842281, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9632352941176471, 'eval_COUNTRY_f1': 0.97119341563786, 'eval_DATE_f1': 0.9179259259259259, 'eval_DRIVERLICENSE_f1': 0.9338036937123885, 'eval_EMAIL_f1': 0.9801098692934268, 'eval_GEOCOORD_f1': 0.9699769053117783, 'eval_GIVENNAME1_f1': 0.8359431462298241, 'eval_GIVENNAME2_f1': 0.7132216014897579, 'eval_IDCARD_f1': 0.9254901960784314, 'eval_IP_f1': 0.9822211960286308, 'eval_LASTNAME1_f1': 0.7888495943415852, 'eval_LASTNAME2_f1': 0.644852354349561, 'eval_LASTNAME3_f1': 0.6237113402061856, 'eval_PASS_f1': 0.9010051781906793, 'eval_PASSPORT_f1': 0.939798643928498, 'eval_POSTCODE_f1': 0.9804024039717795, 'eval_SECADDRESS_f1': 0.9795201872440024, 'eval_SEX_f1': 0.9717821782178218, 'eval_SOCIALNUMBER_f1': 0.9404598152878758, 'eval_STATE_f1': 0.9787659255558332, 'eval_STREET_f1': 0.9679937548790007, 'eval_TEL_f1': 0.9568646783337441, 'eval_T

 29%|██▉       | 13500/46750 [1:13:16<2:30:37,  3.68it/s] 

{'loss': 0.0058, 'grad_norm': 0.6329025626182556, 'learning_rate': 1.7289686274214116e-05, 'epoch': 14.44}


 30%|██▉       | 14000/46750 [1:15:41<2:47:50,  3.25it/s]

{'loss': 0.0056, 'grad_norm': 0.22737400233745575, 'learning_rate': 1.7039261827489452e-05, 'epoch': 14.97}


                                                         
 30%|███       | 14025/46750 [1:16:21<2:14:42,  4.05it/s]

{'eval_loss': 0.053505249321460724, 'eval_BOD_f1': 0.9533032063697009, 'eval_BUILDING_f1': 0.9817902026160554, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9607843137254901, 'eval_COUNTRY_f1': 0.9735516372795969, 'eval_DATE_f1': 0.9167391935015955, 'eval_DRIVERLICENSE_f1': 0.9293924466338259, 'eval_EMAIL_f1': 0.978731484998101, 'eval_GEOCOORD_f1': 0.9744779582366588, 'eval_GIVENNAME1_f1': 0.8375251509054326, 'eval_GIVENNAME2_f1': 0.7153846153846153, 'eval_IDCARD_f1': 0.926720947446336, 'eval_IP_f1': 0.981132075471698, 'eval_LASTNAME1_f1': 0.7912581699346406, 'eval_LASTNAME2_f1': 0.6598079561042524, 'eval_LASTNAME3_f1': 0.6210526315789473, 'eval_PASS_f1': 0.9115479115479115, 'eval_PASSPORT_f1': 0.9350811485642946, 'eval_POSTCODE_f1': 0.9778588174003646, 'eval_SECADDRESS_f1': 0.9755529685681025, 'eval_SEX_f1': 0.973293768545994, 'eval_SOCIALNUMBER_f1': 0.9418031169855988, 'eval_STATE_f1': 0.9783528240855933, 'eval_STREET_f1': 0.9709994821336095, 'eval_TEL_f1': 0.9599213179247603, 'eval_T

 31%|███       | 14500/46750 [1:18:39<2:36:43,  3.43it/s] 

{'loss': 0.0049, 'grad_norm': 0.20782022178173065, 'learning_rate': 1.677976410513221e-05, 'epoch': 15.51}


                                                         
 32%|███▏      | 14960/46750 [1:21:25<2:10:29,  4.06it/s]

{'eval_loss': 0.05221851170063019, 'eval_BOD_f1': 0.9568762068225704, 'eval_BUILDING_f1': 0.9803119406801329, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9621304666503787, 'eval_COUNTRY_f1': 0.9726810673443457, 'eval_DATE_f1': 0.9195804195804196, 'eval_DRIVERLICENSE_f1': 0.9330870279146141, 'eval_EMAIL_f1': 0.9820884146341463, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8327934213805133, 'eval_GIVENNAME2_f1': 0.709796672828096, 'eval_IDCARD_f1': 0.9351921627731726, 'eval_IP_f1': 0.9842810910772076, 'eval_LASTNAME1_f1': 0.7946938775510204, 'eval_LASTNAME2_f1': 0.6527455529775714, 'eval_LASTNAME3_f1': 0.6089108910891089, 'eval_PASS_f1': 0.9032455603184324, 'eval_PASSPORT_f1': 0.9374744585206375, 'eval_POSTCODE_f1': 0.9774669774669775, 'eval_SECADDRESS_f1': 0.9778296382730456, 'eval_SEX_f1': 0.9706390328151988, 'eval_SOCIALNUMBER_f1': 0.9478635829086632, 'eval_STATE_f1': 0.9757064947942489, 'eval_STREET_f1': 0.9691149753438879, 'eval_TEL_f1': 0.9605947955390335, 'eval

 32%|███▏      | 15000/46750 [1:21:38<2:34:27,  3.43it/s] 

{'loss': 0.0044, 'grad_norm': 0.49724847078323364, 'learning_rate': 1.6511527587435736e-05, 'epoch': 16.04}


 33%|███▎      | 15500/46750 [1:24:03<2:33:35,  3.39it/s]

{'loss': 0.0033, 'grad_norm': 0.13960538804531097, 'learning_rate': 1.6234898018587336e-05, 'epoch': 16.58}


                                                         
 34%|███▍      | 15895/46750 [1:26:31<2:20:23,  3.66it/s]

{'eval_loss': 0.05528752878308296, 'eval_BOD_f1': 0.959500215424386, 'eval_BUILDING_f1': 0.9775739041794088, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9607604192054594, 'eval_COUNTRY_f1': 0.9684078823897404, 'eval_DATE_f1': 0.9176811594202898, 'eval_DRIVERLICENSE_f1': 0.9247225025227043, 'eval_EMAIL_f1': 0.9842415037022973, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.8392370572207084, 'eval_GIVENNAME2_f1': 0.7201565557729942, 'eval_IDCARD_f1': 0.9317451314048024, 'eval_IP_f1': 0.9833487511563367, 'eval_LASTNAME1_f1': 0.7915577889447237, 'eval_LASTNAME2_f1': 0.6648005598320503, 'eval_LASTNAME3_f1': 0.6034063260340632, 'eval_PASS_f1': 0.908197727970525, 'eval_PASSPORT_f1': 0.9321672354948806, 'eval_POSTCODE_f1': 0.9791449426485923, 'eval_SECADDRESS_f1': 0.9760932944606414, 'eval_SEX_f1': 0.9735345040811278, 'eval_SOCIALNUMBER_f1': 0.9384346475043698, 'eval_STATE_f1': 0.9801587301587301, 'eval_STREET_f1': 0.9658385093167702, 'eval_TEL_f1': 0.9611097349516969, 'eval_

 34%|███▍      | 16000/46750 [1:27:03<2:17:09,  3.74it/s] 

{'loss': 0.0041, 'grad_norm': 0.33444496989250183, 'learning_rate': 1.5950231961020373e-05, 'epoch': 17.11}


 35%|███▌      | 16500/46750 [1:29:29<2:35:43,  3.24it/s]

{'loss': 0.0028, 'grad_norm': 0.09451606124639511, 'learning_rate': 1.5657896335822147e-05, 'epoch': 17.65}


                                                         
 36%|███▌      | 16830/46750 [1:31:37<2:04:27,  4.01it/s]

{'eval_loss': 0.05723977088928223, 'eval_BOD_f1': 0.9569915712124486, 'eval_BUILDING_f1': 0.980862464914519, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9621674395899439, 'eval_COUNTRY_f1': 0.9714648065948003, 'eval_DATE_f1': 0.9144170017231475, 'eval_DRIVERLICENSE_f1': 0.9398292733708099, 'eval_EMAIL_f1': 0.9853639992396883, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.8362419316280183, 'eval_GIVENNAME2_f1': 0.7147826086956521, 'eval_IDCARD_f1': 0.9297948428383211, 'eval_IP_f1': 0.9838337182448037, 'eval_LASTNAME1_f1': 0.7848639455782312, 'eval_LASTNAME2_f1': 0.6677042801556421, 'eval_LASTNAME3_f1': 0.6284153005464481, 'eval_PASS_f1': 0.9134821155609906, 'eval_PASSPORT_f1': 0.9369587109768379, 'eval_POSTCODE_f1': 0.9781818181818183, 'eval_SECADDRESS_f1': 0.9756380510440836, 'eval_SEX_f1': 0.974308300395257, 'eval_SOCIALNUMBER_f1': 0.9479952362048432, 'eval_STATE_f1': 0.9791252485089463, 'eval_STREET_f1': 0.9671069671069671, 'eval_TEL_f1': 0.9592894152479645, 'eval_

 36%|███▋      | 17000/46750 [1:32:27<2:23:11,  3.46it/s] 

{'loss': 0.0033, 'grad_norm': 0.30449607968330383, 'learning_rate': 1.5358267949789968e-05, 'epoch': 18.18}


 37%|███▋      | 17500/46750 [1:34:53<2:14:43,  3.62it/s]

{'loss': 0.0029, 'grad_norm': 0.29835790395736694, 'learning_rate': 1.5051733009745013e-05, 'epoch': 18.72}


                                                         
 38%|███▊      | 17765/46750 [1:36:42<1:54:27,  4.22it/s]

{'eval_loss': 0.05971790850162506, 'eval_BOD_f1': 0.9549935705100729, 'eval_BUILDING_f1': 0.9817995385798514, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9584042811967891, 'eval_COUNTRY_f1': 0.9746353836398224, 'eval_DATE_f1': 0.9108335753703166, 'eval_DRIVERLICENSE_f1': 0.9336993243243243, 'eval_EMAIL_f1': 0.9845743667872786, 'eval_GEOCOORD_f1': 0.979020979020979, 'eval_GIVENNAME1_f1': 0.8266216542078612, 'eval_GIVENNAME2_f1': 0.7071498530852106, 'eval_IDCARD_f1': 0.9199193696170057, 'eval_IP_f1': 0.9852262234533703, 'eval_LASTNAME1_f1': 0.7917258162644495, 'eval_LASTNAME2_f1': 0.6650602409638554, 'eval_LASTNAME3_f1': 0.6049046321525885, 'eval_PASS_f1': 0.9092028316405047, 'eval_PASSPORT_f1': 0.937344398340249, 'eval_POSTCODE_f1': 0.9780564263322884, 'eval_SECADDRESS_f1': 0.9807130333138515, 'eval_SEX_f1': 0.9754768392370572, 'eval_SOCIALNUMBER_f1': 0.9285851780558229, 'eval_STATE_f1': 0.9803335822753299, 'eval_STREET_f1': 0.9695550351288057, 'eval_TEL_f1': 0.9561534628799203, 'eval_

 39%|███▊      | 18000/46750 [1:37:51<2:26:35,  3.27it/s] 

{'loss': 0.0031, 'grad_norm': 0.0839306116104126, 'learning_rate': 1.4738686624729987e-05, 'epoch': 19.25}


 40%|███▉      | 18500/46750 [1:40:17<2:00:34,  3.90it/s]

{'loss': 0.0025, 'grad_norm': 0.11962549388408661, 'learning_rate': 1.4419532296732271e-05, 'epoch': 19.79}


                                                         
 40%|████      | 18700/46750 [1:41:47<1:51:53,  4.18it/s]

{'eval_loss': 0.056173235177993774, 'eval_BOD_f1': 0.9562231759656653, 'eval_BUILDING_f1': 0.983050847457627, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9663307938068322, 'eval_COUNTRY_f1': 0.9715549936788874, 'eval_DATE_f1': 0.911841722432354, 'eval_DRIVERLICENSE_f1': 0.925739585429664, 'eval_EMAIL_f1': 0.9823361823361824, 'eval_GEOCOORD_f1': 0.9768518518518519, 'eval_GIVENNAME1_f1': 0.8310978638847492, 'eval_GIVENNAME2_f1': 0.719607843137255, 'eval_IDCARD_f1': 0.9313181557298641, 'eval_IP_f1': 0.9865988909426987, 'eval_LASTNAME1_f1': 0.7887037789516457, 'eval_LASTNAME2_f1': 0.6791044776119403, 'eval_LASTNAME3_f1': 0.6323907455012854, 'eval_PASS_f1': 0.9100367197062423, 'eval_PASSPORT_f1': 0.9365609348914858, 'eval_POSTCODE_f1': 0.9817613340281397, 'eval_SECADDRESS_f1': 0.9773913043478262, 'eval_SEX_f1': 0.9727857496288966, 'eval_SOCIALNUMBER_f1': 0.9433222200431456, 'eval_STATE_f1': 0.9803335822753299, 'eval_STREET_f1': 0.972200571577033, 'eval_TEL_f1': 0.9609876543209876, 'eval_TI

 41%|████      | 19000/46750 [1:43:16<2:22:39,  3.24it/s] 

{'loss': 0.0024, 'grad_norm': 0.5601848363876343, 'learning_rate': 1.4094681400588908e-05, 'epoch': 20.32}


 42%|████▏     | 19500/46750 [1:45:40<2:08:22,  3.54it/s]

{'loss': 0.0023, 'grad_norm': 0.21649189293384552, 'learning_rate': 1.376455265374392e-05, 'epoch': 20.86}


                                                         
 42%|████▏     | 19635/46750 [1:46:53<2:08:33,  3.52it/s]

{'eval_loss': 0.06089175492525101, 'eval_BOD_f1': 0.9610836379273274, 'eval_BUILDING_f1': 0.9813060179257364, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9614822038030229, 'eval_COUNTRY_f1': 0.9688972667295005, 'eval_DATE_f1': 0.9120559114735003, 'eval_DRIVERLICENSE_f1': 0.9382921254934553, 'eval_EMAIL_f1': 0.9853528628495338, 'eval_GEOCOORD_f1': 0.97459584295612, 'eval_GIVENNAME1_f1': 0.8419260700389104, 'eval_GIVENNAME2_f1': 0.7157464212678938, 'eval_IDCARD_f1': 0.931740614334471, 'eval_IP_f1': 0.9835914028195054, 'eval_LASTNAME1_f1': 0.791337785873305, 'eval_LASTNAME2_f1': 0.6621108580106303, 'eval_LASTNAME3_f1': 0.6060606060606061, 'eval_PASS_f1': 0.9056372549019609, 'eval_PASSPORT_f1': 0.9368635437881874, 'eval_POSTCODE_f1': 0.9791558103178738, 'eval_SECADDRESS_f1': 0.9790697674418604, 'eval_SEX_f1': 0.9744480277846688, 'eval_SOCIALNUMBER_f1': 0.9417494642509254, 'eval_STATE_f1': 0.9776341948310139, 'eval_STREET_f1': 0.9666580511760146, 'eval_TEL_f1': 0.9561360275998029, 'eval_TI

 43%|████▎     | 20000/46750 [1:48:40<2:11:04,  3.40it/s] 

{'loss': 0.0019, 'grad_norm': 0.22733302414417267, 'learning_rate': 1.3429571576541315e-05, 'epoch': 21.39}


 44%|████▍     | 20500/46750 [1:51:05<1:53:57,  3.84it/s]

{'loss': 0.0019, 'grad_norm': 0.06506683677434921, 'learning_rate': 1.3090169943749475e-05, 'epoch': 21.93}


                                                         
 44%|████▍     | 20570/46750 [1:51:58<1:48:41,  4.01it/s]

{'eval_loss': 0.06329470872879028, 'eval_BOD_f1': 0.96017130620985, 'eval_BUILDING_f1': 0.9807938540332908, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9610326351680468, 'eval_COUNTRY_f1': 0.9695638531534359, 'eval_DATE_f1': 0.9105309474919331, 'eval_DRIVERLICENSE_f1': 0.9315628192032687, 'eval_EMAIL_f1': 0.9866818873668189, 'eval_GEOCOORD_f1': 0.9768518518518519, 'eval_GIVENNAME1_f1': 0.8346091394108096, 'eval_GIVENNAME2_f1': 0.7259786476868326, 'eval_IDCARD_f1': 0.929027412692452, 'eval_IP_f1': 0.9831835982492514, 'eval_LASTNAME1_f1': 0.7854422344477359, 'eval_LASTNAME2_f1': 0.6854020296643248, 'eval_LASTNAME3_f1': 0.6547314578005116, 'eval_PASS_f1': 0.9087009803921569, 'eval_PASSPORT_f1': 0.936266447368421, 'eval_POSTCODE_f1': 0.9789117417339235, 'eval_SECADDRESS_f1': 0.9785008715862871, 'eval_SEX_f1': 0.9716539314764605, 'eval_SOCIALNUMBER_f1': 0.9422275163787968, 'eval_STATE_f1': 0.9779429987608427, 'eval_STREET_f1': 0.9676919100542776, 'eval_TEL_f1': 0.9563521092416483, 'eval_TI

 45%|████▍     | 21000/46750 [1:54:04<2:07:35,  3.36it/s] 

{'loss': 0.0016, 'grad_norm': 0.0810697004199028, 'learning_rate': 1.2746785228023904e-05, 'epoch': 22.46}


 46%|████▌     | 21500/46750 [1:56:28<1:59:34,  3.52it/s]

{'loss': 0.0016, 'grad_norm': 0.13428300619125366, 'learning_rate': 1.239986003602566e-05, 'epoch': 22.99}


                                                         
 46%|████▌     | 21505/46750 [1:57:03<1:58:23,  3.55it/s]

{'eval_loss': 0.06216077879071236, 'eval_BOD_f1': 0.9606536228768006, 'eval_BUILDING_f1': 0.979633401221996, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9647231749142576, 'eval_COUNTRY_f1': 0.9726329034287511, 'eval_DATE_f1': 0.9157530294287363, 'eval_DRIVERLICENSE_f1': 0.9414191419141914, 'eval_EMAIL_f1': 0.9849724177287426, 'eval_GEOCOORD_f1': 0.9699769053117783, 'eval_GIVENNAME1_f1': 0.8385704005756777, 'eval_GIVENNAME2_f1': 0.7240740740740741, 'eval_IDCARD_f1': 0.9308561578241716, 'eval_IP_f1': 0.984302862419206, 'eval_LASTNAME1_f1': 0.7947434292866082, 'eval_LASTNAME2_f1': 0.6769005847953216, 'eval_LASTNAME3_f1': 0.6415094339622642, 'eval_PASS_f1': 0.8775015160703458, 'eval_PASSPORT_f1': 0.9365274365274366, 'eval_POSTCODE_f1': 0.9787344398340249, 'eval_SECADDRESS_f1': 0.9790697674418604, 'eval_SEX_f1': 0.9726533628972653, 'eval_SOCIALNUMBER_f1': 0.9389797882579403, 'eval_STATE_f1': 0.9798857710454433, 'eval_STREET_f1': 0.9702457956015524, 'eval_TEL_f1': 0.9572311495673671, 'eval_

 47%|████▋     | 22000/46750 [1:59:27<1:51:41,  3.69it/s] 

{'loss': 0.0016, 'grad_norm': 0.03914812579751015, 'learning_rate': 1.2049841537922307e-05, 'epoch': 23.53}


                                                         
 48%|████▊     | 22440/46750 [2:02:07<1:45:19,  3.85it/s]

{'eval_loss': 0.06269590556621552, 'eval_BOD_f1': 0.9593810444874273, 'eval_BUILDING_f1': 0.978952772073922, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9700833742030407, 'eval_COUNTRY_f1': 0.9746675110829639, 'eval_DATE_f1': 0.911266783420899, 'eval_DRIVERLICENSE_f1': 0.9330862672431542, 'eval_EMAIL_f1': 0.9861084681255946, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.8451739343459089, 'eval_GIVENNAME2_f1': 0.7331433998100664, 'eval_IDCARD_f1': 0.9241302738712064, 'eval_IP_f1': 0.9861367837338263, 'eval_LASTNAME1_f1': 0.7974921630094044, 'eval_LASTNAME2_f1': 0.6800308404009252, 'eval_LASTNAME3_f1': 0.6461538461538462, 'eval_PASS_f1': 0.9085328422344997, 'eval_PASSPORT_f1': 0.9308045018757816, 'eval_POSTCODE_f1': 0.9794751883606132, 'eval_SECADDRESS_f1': 0.9801864801864801, 'eval_SEX_f1': 0.9737883283877349, 'eval_SOCIALNUMBER_f1': 0.9404808829325976, 'eval_STATE_f1': 0.9769288017861573, 'eval_STREET_f1': 0.9691789691789691, 'eval_TEL_f1': 0.9574046557701833, 'eval_

 48%|████▊     | 22500/46750 [2:02:26<2:07:27,  3.17it/s] 

{'loss': 0.0015, 'grad_norm': 0.031180569902062416, 'learning_rate': 1.169718089100669e-05, 'epoch': 24.06}


 49%|████▉     | 23000/46750 [2:04:50<2:07:15,  3.11it/s]

{'loss': 0.0012, 'grad_norm': 0.03342045098543167, 'learning_rate': 1.1342332658176556e-05, 'epoch': 24.6}


                                                         
 50%|█████     | 23375/46750 [2:07:12<1:35:05,  4.10it/s]

{'eval_loss': 0.06286289542913437, 'eval_BOD_f1': 0.95706311721769, 'eval_BUILDING_f1': 0.9782775364170713, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9659397206567018, 'eval_COUNTRY_f1': 0.9744398863994951, 'eval_DATE_f1': 0.9123621590249564, 'eval_DRIVERLICENSE_f1': 0.9414191419141914, 'eval_EMAIL_f1': 0.9847619047619048, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8414779499404051, 'eval_GIVENNAME2_f1': 0.7303689687795648, 'eval_IDCARD_f1': 0.9275145158269338, 'eval_IP_f1': 0.9836216839677048, 'eval_LASTNAME1_f1': 0.7966893039049234, 'eval_LASTNAME2_f1': 0.6700143472022956, 'eval_LASTNAME3_f1': 0.6440677966101696, 'eval_PASS_f1': 0.9113224915618288, 'eval_PASSPORT_f1': 0.9384551838083437, 'eval_POSTCODE_f1': 0.9794002607561929, 'eval_SECADDRESS_f1': 0.980209545983702, 'eval_SEX_f1': 0.9754768392370572, 'eval_SOCIALNUMBER_f1': 0.9416812951043494, 'eval_STATE_f1': 0.9800598205383848, 'eval_STREET_f1': 0.9702303908879109, 'eval_TEL_f1': 0.9603590127150337, 'eval_T

 50%|█████     | 23500/46750 [2:07:49<1:47:24,  3.61it/s] 

{'loss': 0.0013, 'grad_norm': 0.015984097495675087, 'learning_rate': 1.0985754222024437e-05, 'epoch': 25.13}


 51%|█████▏    | 24000/46750 [2:10:14<1:49:25,  3.46it/s]

{'loss': 0.0009, 'grad_norm': 0.3857351541519165, 'learning_rate': 1.0627905195293135e-05, 'epoch': 25.67}


                                                         
 52%|█████▏    | 24310/46750 [2:12:16<1:35:35,  3.91it/s]

{'eval_loss': 0.06428779661655426, 'eval_BOD_f1': 0.9589864719776681, 'eval_BUILDING_f1': 0.9780163599182005, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9646255184191266, 'eval_COUNTRY_f1': 0.9720740508315029, 'eval_DATE_f1': 0.9087739686228937, 'eval_DRIVERLICENSE_f1': 0.9413471502590672, 'eval_EMAIL_f1': 0.9845978322875071, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.844027877913963, 'eval_GIVENNAME2_f1': 0.718552036199095, 'eval_IDCARD_f1': 0.9282242990654206, 'eval_IP_f1': 0.984750462107209, 'eval_LASTNAME1_f1': 0.79983474488742, 'eval_LASTNAME2_f1': 0.6612665684830633, 'eval_LASTNAME3_f1': 0.6206896551724137, 'eval_PASS_f1': 0.9059724349157734, 'eval_PASSPORT_f1': 0.9366111336854936, 'eval_POSTCODE_f1': 0.9804941482444733, 'eval_SECADDRESS_f1': 0.9773387565368972, 'eval_SEX_f1': 0.9729863692688971, 'eval_SOCIALNUMBER_f1': 0.9407114624505929, 'eval_STATE_f1': 0.9769288017861573, 'eval_STREET_f1': 0.9694142042509072, 'eval_TEL_f1': 0.9573155687145323, 'eval_TIM

 52%|█████▏    | 24500/46750 [2:13:12<1:44:23,  3.55it/s] 

{'loss': 0.0009, 'grad_norm': 0.020713578909635544, 'learning_rate': 1.026924682845663e-05, 'epoch': 26.2}


 53%|█████▎    | 25000/46750 [2:15:38<1:44:58,  3.45it/s]

{'loss': 0.0011, 'grad_norm': 0.08223766833543777, 'learning_rate': 9.910241415190022e-06, 'epoch': 26.74}


                                                         
 54%|█████▍    | 25245/46750 [2:17:22<1:16:44,  4.67it/s]

{'eval_loss': 0.06673801690340042, 'eval_BOD_f1': 0.9581716257007331, 'eval_BUILDING_f1': 0.9803621525121142, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.963578587142508, 'eval_COUNTRY_f1': 0.9747315224257739, 'eval_DATE_f1': 0.9135229251305863, 'eval_DRIVERLICENSE_f1': 0.937269372693727, 'eval_EMAIL_f1': 0.9838679066236476, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.8477366255144032, 'eval_GIVENNAME2_f1': 0.7340529931305201, 'eval_IDCARD_f1': 0.9294478527607363, 'eval_IP_f1': 0.984060984060984, 'eval_LASTNAME1_f1': 0.7937422807739811, 'eval_LASTNAME2_f1': 0.6765578635014836, 'eval_LASTNAME3_f1': 0.6414141414141414, 'eval_PASS_f1': 0.900459418070444, 'eval_PASSPORT_f1': 0.9353658536585365, 'eval_POSTCODE_f1': 0.9771784232365145, 'eval_SECADDRESS_f1': 0.9756380510440836, 'eval_SEX_f1': 0.9750555692763645, 'eval_SOCIALNUMBER_f1': 0.9399610136452241, 'eval_STATE_f1': 0.9774361517480784, 'eval_STREET_f1': 0.9667877529839127, 'eval_TEL_f1': 0.9605556933763335, 'eval_TI

 55%|█████▍    | 25500/46750 [2:18:36<1:39:19,  3.57it/s] 

{'loss': 0.001, 'grad_norm': 0.06282424926757812, 'learning_rate': 9.551351696494854e-06, 'epoch': 27.27}


 56%|█████▌    | 26000/46750 [2:21:00<1:35:15,  3.63it/s]

{'loss': 0.0009, 'grad_norm': 0.013104872778058052, 'learning_rate': 9.19304026424783e-06, 'epoch': 27.81}


                                                         
 56%|█████▌    | 26180/46750 [2:22:26<1:37:39,  3.51it/s]

{'eval_loss': 0.06696619838476181, 'eval_BOD_f1': 0.9560557341907824, 'eval_BUILDING_f1': 0.9825997952917094, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9647921760391198, 'eval_COUNTRY_f1': 0.9717159019484601, 'eval_DATE_f1': 0.9131964809384164, 'eval_DRIVERLICENSE_f1': 0.9404166666666667, 'eval_EMAIL_f1': 0.9845919726079514, 'eval_GEOCOORD_f1': 0.9631336405529954, 'eval_GIVENNAME1_f1': 0.8384691848906561, 'eval_GIVENNAME2_f1': 0.7156673114119922, 'eval_IDCARD_f1': 0.9280923994038748, 'eval_IP_f1': 0.9834024896265561, 'eval_LASTNAME1_f1': 0.7938227035699961, 'eval_LASTNAME2_f1': 0.6788856304985337, 'eval_LASTNAME3_f1': 0.6264367816091954, 'eval_PASS_f1': 0.9107635694572217, 'eval_PASSPORT_f1': 0.9396604622622212, 'eval_POSTCODE_f1': 0.9809548656404905, 'eval_SECADDRESS_f1': 0.9761766414875074, 'eval_SEX_f1': 0.9730264785944073, 'eval_SOCIALNUMBER_f1': 0.9413154785348563, 'eval_STATE_f1': 0.9796019900497512, 'eval_STREET_f1': 0.9697126585555269, 'eval_TEL_f1': 0.9608470819995075, 'eva

 57%|█████▋    | 26500/46750 [2:24:00<1:31:51,  3.67it/s] 

{'loss': 0.0008, 'grad_norm': 0.7304526567459106, 'learning_rate': 8.835768964941773e-06, 'epoch': 28.34}


 58%|█████▊    | 27000/46750 [2:26:25<1:30:42,  3.63it/s]

{'loss': 0.0008, 'grad_norm': 0.15264703333377838, 'learning_rate': 8.479998304387329e-06, 'epoch': 28.88}


                                                         
 58%|█████▊    | 27115/46750 [2:27:31<1:20:07,  4.08it/s]

{'eval_loss': 0.0670015960931778, 'eval_BOD_f1': 0.9597257926306769, 'eval_BUILDING_f1': 0.9792573623559538, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9669036528560921, 'eval_COUNTRY_f1': 0.9735015772870661, 'eval_DATE_f1': 0.9153237620861413, 'eval_DRIVERLICENSE_f1': 0.9396891191709844, 'eval_EMAIL_f1': 0.9859315589353613, 'eval_GEOCOORD_f1': 0.9767441860465117, 'eval_GIVENNAME1_f1': 0.8461160275319567, 'eval_GIVENNAME2_f1': 0.7269406392694064, 'eval_IDCARD_f1': 0.9308651303362298, 'eval_IP_f1': 0.985654789449329, 'eval_LASTNAME1_f1': 0.7962395258532597, 'eval_LASTNAME2_f1': 0.6767830045523521, 'eval_LASTNAME3_f1': 0.6228571428571428, 'eval_PASS_f1': 0.9121081745543946, 'eval_PASSPORT_f1': 0.9378206443669197, 'eval_POSTCODE_f1': 0.9827676240208878, 'eval_SECADDRESS_f1': 0.9790209790209791, 'eval_SEX_f1': 0.9745364647713225, 'eval_SOCIALNUMBER_f1': 0.9449433814916048, 'eval_STATE_f1': 0.9790940766550523, 'eval_STREET_f1': 0.9713104161281986, 'eval_TEL_f1': 0.9606299212598426, 'eval_

 59%|█████▉    | 27500/46750 [2:29:23<1:25:01,  3.77it/s] 

{'loss': 0.0008, 'grad_norm': 0.021056482568383217, 'learning_rate': 8.126186854142752e-06, 'epoch': 29.41}


 60%|█████▉    | 28000/46750 [2:31:49<1:30:19,  3.46it/s]

{'loss': 0.0006, 'grad_norm': 0.07860207557678223, 'learning_rate': 7.774790660436857e-06, 'epoch': 29.95}


                                                         
 60%|██████    | 28050/46750 [2:32:37<1:25:34,  3.64it/s]

{'eval_loss': 0.06624649465084076, 'eval_BOD_f1': 0.9609155689915785, 'eval_BUILDING_f1': 0.9815762538382805, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9635429410325422, 'eval_COUNTRY_f1': 0.9738582677165355, 'eval_DATE_f1': 0.9128175519630485, 'eval_DRIVERLICENSE_f1': 0.937269372693727, 'eval_EMAIL_f1': 0.9853528628495338, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8466650258429732, 'eval_GIVENNAME2_f1': 0.7245508982035928, 'eval_IDCARD_f1': 0.9327423395644633, 'eval_IP_f1': 0.9861239592969473, 'eval_LASTNAME1_f1': 0.7975733063700708, 'eval_LASTNAME2_f1': 0.6863636363636363, 'eval_LASTNAME3_f1': 0.6614173228346456, 'eval_PASS_f1': 0.9189522342064714, 'eval_PASSPORT_f1': 0.9349209607883392, 'eval_POSTCODE_f1': 0.9825019587359625, 'eval_SECADDRESS_f1': 0.9796156086196854, 'eval_SEX_f1': 0.9744986382768013, 'eval_SOCIALNUMBER_f1': 0.9430453312669509, 'eval_STATE_f1': 0.9810756972111553, 'eval_STREET_f1': 0.9742923915866009, 'eval_TEL_f1': 0.9589377919842637, 'eval

 61%|██████    | 28501/46750 [2:34:49<1:15:48,  4.01it/s] 

{'loss': 0.0005, 'grad_norm': 0.003951985388994217, 'learning_rate': 7.426262656346979e-06, 'epoch': 30.48}


                                                         
 62%|██████▏   | 28985/46750 [2:37:42<1:14:27,  3.98it/s]

{'eval_loss': 0.06997386366128922, 'eval_BOD_f1': 0.9566694987255735, 'eval_BUILDING_f1': 0.9818274891220886, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9650452212173062, 'eval_COUNTRY_f1': 0.9720037747719408, 'eval_DATE_f1': 0.9068873780668046, 'eval_DRIVERLICENSE_f1': 0.9370400654129191, 'eval_EMAIL_f1': 0.9842355175688509, 'eval_GEOCOORD_f1': 0.9699769053117783, 'eval_GIVENNAME1_f1': 0.8428504011326098, 'eval_GIVENNAME2_f1': 0.730186999109528, 'eval_IDCARD_f1': 0.9318010550113037, 'eval_IP_f1': 0.9852125693160815, 'eval_LASTNAME1_f1': 0.7882229232386962, 'eval_LASTNAME2_f1': 0.6774941995359629, 'eval_LASTNAME3_f1': 0.65, 'eval_PASS_f1': 0.9097030915212733, 'eval_PASSPORT_f1': 0.9359868691013542, 'eval_POSTCODE_f1': 0.9804534792806879, 'eval_SECADDRESS_f1': 0.9767711962833914, 'eval_SEX_f1': 0.9729330708661418, 'eval_SOCIALNUMBER_f1': 0.946329913180742, 'eval_STATE_f1': 0.9776896380763511, 'eval_STREET_f1': 0.9717250324254215, 'eval_TEL_f1': 0.9600788565795958, 'eval_TIME_f1': 0.96

 62%|██████▏   | 29000/46750 [2:37:48<1:40:05,  2.96it/s] 

{'loss': 0.0007, 'grad_norm': 0.004321034532040358, 'learning_rate': 7.081052077989668e-06, 'epoch': 31.02}


 63%|██████▎   | 29500/46750 [2:40:13<1:20:28,  3.57it/s]

{'loss': 0.0006, 'grad_norm': 0.014548027887940407, 'learning_rate': 6.7396038854765825e-06, 'epoch': 31.55}


                                                         
 64%|██████▍   | 29920/46750 [2:42:47<1:07:46,  4.14it/s]

{'eval_loss': 0.06746794283390045, 'eval_BOD_f1': 0.9595698924731183, 'eval_BUILDING_f1': 0.9818274891220886, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9655003670173722, 'eval_COUNTRY_f1': 0.9713385826771653, 'eval_DATE_f1': 0.9146198830409357, 'eval_DRIVERLICENSE_f1': 0.9413218035824583, 'eval_EMAIL_f1': 0.9847850893875998, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8445914029286727, 'eval_GIVENNAME2_f1': 0.7179006560449859, 'eval_IDCARD_f1': 0.9327794561933533, 'eval_IP_f1': 0.9877286408890946, 'eval_LASTNAME1_f1': 0.792199622562382, 'eval_LASTNAME2_f1': 0.671664167916042, 'eval_LASTNAME3_f1': 0.649859943977591, 'eval_PASS_f1': 0.9169475942384309, 'eval_PASSPORT_f1': 0.9371213703781074, 'eval_POSTCODE_f1': 0.9812304483837331, 'eval_SECADDRESS_f1': 0.9773387565368972, 'eval_SEX_f1': 0.973066468989375, 'eval_SOCIALNUMBER_f1': 0.9409038238702202, 'eval_STATE_f1': 0.9804116042648153, 'eval_STREET_f1': 0.9724675324675324, 'eval_TEL_f1': 0.9627606752730884, 'eval_TI

 64%|██████▍   | 30000/46750 [2:43:11<1:28:49,  3.14it/s] 

{'loss': 0.0004, 'grad_norm': 0.24949930608272552, 'learning_rate': 6.4023581893819345e-06, 'epoch': 32.09}


 65%|██████▌   | 30500/46750 [2:45:36<1:16:19,  3.55it/s]

{'loss': 0.0005, 'grad_norm': 0.001148969866335392, 'learning_rate': 6.069749683460765e-06, 'epoch': 32.62}


                                                         
 66%|██████▌   | 30855/46750 [2:47:51<1:09:52,  3.79it/s]

{'eval_loss': 0.06977463513612747, 'eval_BOD_f1': 0.9581993569131831, 'eval_BUILDING_f1': 0.9813537675606641, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.962545899632803, 'eval_COUNTRY_f1': 0.9727016002510198, 'eval_DATE_f1': 0.910781749491427, 'eval_DRIVERLICENSE_f1': 0.9387839800788544, 'eval_EMAIL_f1': 0.9846153846153846, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.839645843580915, 'eval_GIVENNAME2_f1': 0.7124378109452737, 'eval_IDCARD_f1': 0.9324273310683276, 'eval_IP_f1': 0.9861239592969473, 'eval_LASTNAME1_f1': 0.7930144870013892, 'eval_LASTNAME2_f1': 0.6736526946107785, 'eval_LASTNAME3_f1': 0.6374269005847953, 'eval_PASS_f1': 0.9153374233128835, 'eval_PASSPORT_f1': 0.9378438964744243, 'eval_POSTCODE_f1': 0.980473834938818, 'eval_SECADDRESS_f1': 0.9796629866356769, 'eval_SEX_f1': 0.9742701632855023, 'eval_SOCIALNUMBER_f1': 0.9409241567557028, 'eval_STATE_f1': 0.9786176031824962, 'eval_STREET_f1': 0.9726633689143451, 'eval_TEL_f1': 0.9657568238213399, 'eval_TI

 66%|██████▋   | 31000/46750 [2:48:34<1:08:35,  3.83it/s] 

{'loss': 0.0005, 'grad_norm': 0.0814817026257515, 'learning_rate': 5.742207084349274e-06, 'epoch': 33.16}


 67%|██████▋   | 31500/46750 [2:50:57<1:16:45,  3.31it/s]

{'loss': 0.0004, 'grad_norm': 0.14676699042320251, 'learning_rate': 5.420152578969327e-06, 'epoch': 33.69}


                                                         
 68%|██████▊   | 31790/46750 [2:52:55<1:00:17,  4.14it/s]

{'eval_loss': 0.06713838130235672, 'eval_BOD_f1': 0.9619436680283809, 'eval_BUILDING_f1': 0.9825997952917094, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9664954756664221, 'eval_COUNTRY_f1': 0.9731945758435825, 'eval_DATE_f1': 0.911841722432354, 'eval_DRIVERLICENSE_f1': 0.9382716049382717, 'eval_EMAIL_f1': 0.983837231412816, 'eval_GEOCOORD_f1': 0.979020979020979, 'eval_GIVENNAME1_f1': 0.8444224585703686, 'eval_GIVENNAME2_f1': 0.7284522706209454, 'eval_IDCARD_f1': 0.9302673396896616, 'eval_IP_f1': 0.9856614246068455, 'eval_LASTNAME1_f1': 0.7987878787878787, 'eval_LASTNAME2_f1': 0.6716085989621942, 'eval_LASTNAME3_f1': 0.6612903225806452, 'eval_PASS_f1': 0.9186367823150139, 'eval_PASSPORT_f1': 0.9369964883288578, 'eval_POSTCODE_f1': 0.9822916666666667, 'eval_SECADDRESS_f1': 0.9773650609402206, 'eval_SEX_f1': 0.9755011135857461, 'eval_SOCIALNUMBER_f1': 0.9457914449043958, 'eval_STATE_f1': 0.9798457327693456, 'eval_STREET_f1': 0.9717689717689717, 'eval_TEL_f1': 0.9621005697299975, 'eval_T

 68%|██████▊   | 32000/46750 [2:53:58<1:17:55,  3.15it/s] 

{'loss': 0.0004, 'grad_norm': 0.12206078320741653, 'learning_rate': 5.10400128034948e-06, 'epoch': 34.22}


 70%|██████▉   | 32500/46750 [2:56:21<1:05:46,  3.61it/s]

{'loss': 0.0003, 'grad_norm': 0.017783967778086662, 'learning_rate': 4.794160692563917e-06, 'epoch': 34.76}


                                                         
 70%|███████   | 32725/46750 [2:57:59<57:35,  4.06it/s]

{'eval_loss': 0.0697576105594635, 'eval_BOD_f1': 0.9615301955727487, 'eval_BUILDING_f1': 0.9819016059138415, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9647231749142576, 'eval_COUNTRY_f1': 0.9741976085588421, 'eval_DATE_f1': 0.9121071012805588, 'eval_DRIVERLICENSE_f1': 0.9371171906900776, 'eval_EMAIL_f1': 0.9851767388825541, 'eval_GEOCOORD_f1': 0.97459584295612, 'eval_GIVENNAME1_f1': 0.8433048433048433, 'eval_GIVENNAME2_f1': 0.7271111111111112, 'eval_IDCARD_f1': 0.9325971058644326, 'eval_IP_f1': 0.984984984984985, 'eval_LASTNAME1_f1': 0.7956173619890434, 'eval_LASTNAME2_f1': 0.6776119402985075, 'eval_LASTNAME3_f1': 0.6478873239436619, 'eval_PASS_f1': 0.9119804400977997, 'eval_PASSPORT_f1': 0.9364303178484108, 'eval_POSTCODE_f1': 0.9820265694191196, 'eval_SECADDRESS_f1': 0.9779069767441859, 'eval_SEX_f1': 0.9762611275964392, 'eval_SOCIALNUMBER_f1': 0.9428684003152088, 'eval_STATE_f1': 0.9783850931677018, 'eval_STREET_f1': 0.972229431611731, 'eval_TEL_f1': 0.9641000247586036, 'eval_TIM

 71%|███████   | 33000/46750 [2:59:19<1:14:59,  3.06it/s] 

{'loss': 0.0003, 'grad_norm': 0.06561919301748276, 'learning_rate': 4.491030185478976e-06, 'epoch': 35.29}


 72%|███████▏  | 33500/46750 [3:01:45<1:00:12,  3.67it/s]

{'loss': 0.0003, 'grad_norm': 0.002758509712293744, 'learning_rate': 4.195000479984264e-06, 'epoch': 35.83}


                                                         
 72%|███████▏  | 33660/46750 [3:03:04<56:41,  3.85it/s]

{'eval_loss': 0.07023075968027115, 'eval_BOD_f1': 0.9582616179001722, 'eval_BUILDING_f1': 0.9800204918032787, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9623471882640586, 'eval_COUNTRY_f1': 0.974763406940063, 'eval_DATE_f1': 0.911046511627907, 'eval_DRIVERLICENSE_f1': 0.941128205128205, 'eval_EMAIL_f1': 0.9848082035700721, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.8420542635658914, 'eval_GIVENNAME2_f1': 0.7235996326905417, 'eval_IDCARD_f1': 0.9323451159255036, 'eval_IP_f1': 0.9863646868500116, 'eval_LASTNAME1_f1': 0.7939431143851033, 'eval_LASTNAME2_f1': 0.6779400461183704, 'eval_LASTNAME3_f1': 0.6630727762803235, 'eval_PASS_f1': 0.9174816625916871, 'eval_PASSPORT_f1': 0.9412248253185369, 'eval_POSTCODE_f1': 0.9815056004167753, 'eval_SECADDRESS_f1': 0.9756380510440836, 'eval_SEX_f1': 0.9733069698467622, 'eval_SOCIALNUMBER_f1': 0.9472243940578577, 'eval_STATE_f1': 0.9783312577833125, 'eval_STREET_f1': 0.9717396940627431, 'eval_TEL_f1': 0.9632007903185972, 'eval_T

 73%|███████▎  | 34000/46750 [3:04:43<1:02:04,  3.42it/s] 

{'loss': 0.0003, 'grad_norm': 0.033836595714092255, 'learning_rate': 3.90645314437192e-06, 'epoch': 36.36}


 74%|███████▍  | 34500/46750 [3:07:09<1:03:12,  3.23it/s]

{'loss': 0.0003, 'grad_norm': 0.2316432148218155, 'learning_rate': 3.625760102513103e-06, 'epoch': 36.9}


                                                         
 74%|███████▍  | 34595/46750 [3:08:10<49:30,  4.09it/s]

{'eval_loss': 0.0694008320569992, 'eval_BOD_f1': 0.959724316174887, 'eval_BUILDING_f1': 0.98257303946694, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9666503187837177, 'eval_COUNTRY_f1': 0.9744721084147495, 'eval_DATE_f1': 0.9136018680677175, 'eval_DRIVERLICENSE_f1': 0.9412489728841413, 'eval_EMAIL_f1': 0.9838495154854645, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8484701311316174, 'eval_GIVENNAME2_f1': 0.7295825771324864, 'eval_IDCARD_f1': 0.9357798165137615, 'eval_IP_f1': 0.9870490286771508, 'eval_LASTNAME1_f1': 0.7996681874740772, 'eval_LASTNAME2_f1': 0.6771771771771771, 'eval_LASTNAME3_f1': 0.6595174262734586, 'eval_PASS_f1': 0.9179219182293268, 'eval_PASSPORT_f1': 0.9416103257529196, 'eval_POSTCODE_f1': 0.9796663190823774, 'eval_SECADDRESS_f1': 0.9790697674418604, 'eval_SEX_f1': 0.9746772591857, 'eval_SOCIALNUMBER_f1': 0.9458722741433022, 'eval_STATE_f1': 0.9779101514023331, 'eval_STREET_f1': 0.9719917012448133, 'eval_TEL_f1': 0.966923650833126, 'eval_TIME_f

 75%|███████▍  | 35000/46750 [3:10:07<58:09,  3.37it/s]   

{'loss': 0.0002, 'grad_norm': 0.054451752454042435, 'learning_rate': 3.3532831544657464e-06, 'epoch': 37.43}


 76%|███████▌  | 35500/46750 [3:12:33<55:51,  3.36it/s]  

{'loss': 0.0003, 'grad_norm': 0.0433049350976944, 'learning_rate': 3.089373510131354e-06, 'epoch': 37.97}


                                                       
 76%|███████▌  | 35530/46750 [3:13:15<46:32,  4.02it/s]

{'eval_loss': 0.06989865005016327, 'eval_BOD_f1': 0.9604495353360709, 'eval_BUILDING_f1': 0.9813537675606641, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9659230203481246, 'eval_COUNTRY_f1': 0.9739847715736041, 'eval_DATE_f1': 0.9199417758369723, 'eval_DRIVERLICENSE_f1': 0.9373464373464373, 'eval_EMAIL_f1': 0.9857441551035925, 'eval_GEOCOORD_f1': 0.9791183294663574, 'eval_GIVENNAME1_f1': 0.8474903474903476, 'eval_GIVENNAME2_f1': 0.7327502429543247, 'eval_IDCARD_f1': 0.9319470699432892, 'eval_IP_f1': 0.9845229845229846, 'eval_LASTNAME1_f1': 0.8001644736842106, 'eval_LASTNAME2_f1': 0.6840148698884758, 'eval_LASTNAME3_f1': 0.6505376344086022, 'eval_PASS_f1': 0.9156700398650721, 'eval_PASSPORT_f1': 0.9411283728536385, 'eval_POSTCODE_f1': 0.9812597605413848, 'eval_SECADDRESS_f1': 0.9779069767441859, 'eval_SEX_f1': 0.9744860044587565, 'eval_SOCIALNUMBER_f1': 0.9477817039654497, 'eval_STATE_f1': 0.9798256537982565, 'eval_STREET_f1': 0.9696969696969697, 'eval_TEL_f1': 0.962834489593657, 'eval

 77%|███████▋  | 36000/46750 [3:15:33<55:09,  3.25it/s]   

{'loss': 0.0002, 'grad_norm': 0.004072383977472782, 'learning_rate': 2.834371336562077e-06, 'epoch': 38.5}


                                                         
 78%|███████▊  | 36465/46750 [3:18:21<43:22,  3.95it/s]

{'eval_loss': 0.07087461650371552, 'eval_BOD_f1': 0.9597935927757472, 'eval_BUILDING_f1': 0.981363288230789, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.964486896889542, 'eval_COUNTRY_f1': 0.9725465446513095, 'eval_DATE_f1': 0.9167883211678832, 'eval_DRIVERLICENSE_f1': 0.9414674361088212, 'eval_EMAIL_f1': 0.9847966552641582, 'eval_GEOCOORD_f1': 0.9768518518518519, 'eval_GIVENNAME1_f1': 0.8525059665871121, 'eval_GIVENNAME2_f1': 0.7327823691460056, 'eval_IDCARD_f1': 0.9348908464189966, 'eval_IP_f1': 0.9831835982492514, 'eval_LASTNAME1_f1': 0.8030176026823135, 'eval_LASTNAME2_f1': 0.6878228782287823, 'eval_LASTNAME3_f1': 0.6590257879656161, 'eval_PASS_f1': 0.9112063686466626, 'eval_PASSPORT_f1': 0.9402893825147747, 'eval_POSTCODE_f1': 0.9825475384214639, 'eval_SECADDRESS_f1': 0.9790454016298021, 'eval_SEX_f1': 0.9750061865874784, 'eval_SOCIALNUMBER_f1': 0.9471631897055957, 'eval_STATE_f1': 0.9796019900497512, 'eval_STREET_f1': 0.9694458829621957, 'eval_TEL_f1': 0.9670384138785626, 'eval_

 78%|███████▊  | 36500/46750 [3:18:32<49:54,  3.42it/s]   

{'loss': 0.0002, 'grad_norm': 0.0024651384446769953, 'learning_rate': 2.5886053195014537e-06, 'epoch': 39.04}


 79%|███████▉  | 37000/46750 [3:20:58<49:57,  3.25it/s]  

{'loss': 0.0002, 'grad_norm': 0.039074864238500595, 'learning_rate': 2.3523922397240163e-06, 'epoch': 39.57}


                                                       
 80%|████████  | 37400/46750 [3:23:29<37:20,  4.17it/s]

{'eval_loss': 0.07152894139289856, 'eval_BOD_f1': 0.9593810444874273, 'eval_BUILDING_f1': 0.9816044966785897, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9621674395899439, 'eval_COUNTRY_f1': 0.9747474747474747, 'eval_DATE_f1': 0.9152046783625731, 'eval_DRIVERLICENSE_f1': 0.9424714434060228, 'eval_EMAIL_f1': 0.9840425531914894, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8511761881901103, 'eval_GIVENNAME2_f1': 0.7248826291079813, 'eval_IDCARD_f1': 0.9310409269295459, 'eval_IP_f1': 0.9854537058416071, 'eval_LASTNAME1_f1': 0.7990105132962275, 'eval_LASTNAME2_f1': 0.6853146853146853, 'eval_LASTNAME3_f1': 0.6702702702702703, 'eval_PASS_f1': 0.9113768782582029, 'eval_PASSPORT_f1': 0.9410071942446043, 'eval_POSTCODE_f1': 0.9820452771272443, 'eval_SECADDRESS_f1': 0.9784758580570099, 'eval_SEX_f1': 0.9755253399258343, 'eval_SOCIALNUMBER_f1': 0.9468334636434713, 'eval_STATE_f1': 0.9791148682247638, 'eval_STREET_f1': 0.9712211563391238, 'eval_TEL_f1': 0.9671641791044777, 'eva

 80%|████████  | 37500/46750 [3:23:58<39:54,  3.86it/s]   

{'loss': 0.0002, 'grad_norm': 0.02257128432393074, 'learning_rate': 2.12603656471988e-06, 'epoch': 40.11}


 81%|████████▏ | 38000/46750 [3:26:24<45:49,  3.18it/s]

{'loss': 0.0002, 'grad_norm': 0.03793947026133537, 'learning_rate': 1.9098300562505266e-06, 'epoch': 40.64}


                                                       
 82%|████████▏ | 38335/46750 [3:28:36<34:24,  4.08it/s]

{'eval_loss': 0.07147062569856644, 'eval_BOD_f1': 0.9605518430696269, 'eval_BUILDING_f1': 0.9825997952917094, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.9640322975287496, 'eval_COUNTRY_f1': 0.9740834386852086, 'eval_DATE_f1': 0.9150174621653085, 'eval_DRIVERLICENSE_f1': 0.9404000824912353, 'eval_EMAIL_f1': 0.9851654621529099, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8488344148041337, 'eval_GIVENNAME2_f1': 0.7277829747427501, 'eval_IDCARD_f1': 0.9342982289087792, 'eval_IP_f1': 0.9852262234533703, 'eval_LASTNAME1_f1': 0.7983471074380166, 'eval_LASTNAME2_f1': 0.6829640947288006, 'eval_LASTNAME3_f1': 0.6592178770949721, 'eval_PASS_f1': 0.9192508443352778, 'eval_PASSPORT_f1': 0.9409836065573771, 'eval_POSTCODE_f1': 0.9812695109261186, 'eval_SECADDRESS_f1': 0.9761766414875074, 'eval_SEX_f1': 0.9742956005931784, 'eval_SOCIALNUMBER_f1': 0.9489736070381232, 'eval_STATE_f1': 0.9788819875776398, 'eval_STREET_f1': 0.9719771665801765, 'eval_TEL_f1': 0.9615194869264924, 'eva

 82%|████████▏ | 38500/46750 [3:29:24<37:10,  3.70it/s]   

{'loss': 0.0002, 'grad_norm': 0.003088444471359253, 'learning_rate': 1.7040513942816905e-06, 'epoch': 41.18}


 83%|████████▎ | 39000/46750 [3:31:50<37:35,  3.44it/s]

{'loss': 0.0002, 'grad_norm': 0.20396578311920166, 'learning_rate': 1.5089658177780653e-06, 'epoch': 41.71}


                                                       
 84%|████████▍ | 39270/46750 [3:33:43<31:35,  3.95it/s]

{'eval_loss': 0.07122096419334412, 'eval_BOD_f1': 0.959724316174887, 'eval_BUILDING_f1': 0.9816232771822359, 'eval_CARDISSUER_f1': 0.0, 'eval_CITY_f1': 0.965906303654648, 'eval_COUNTRY_f1': 0.9747315224257739, 'eval_DATE_f1': 0.9183971921614507, 'eval_DRIVERLICENSE_f1': 0.9412008281573498, 'eval_EMAIL_f1': 0.9840425531914894, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8500595947556615, 'eval_GIVENNAME2_f1': 0.7168059424326834, 'eval_IDCARD_f1': 0.9311573813543425, 'eval_IP_f1': 0.9870550161812297, 'eval_LASTNAME1_f1': 0.8000832986255727, 'eval_LASTNAME2_f1': 0.6858029480217223, 'eval_LASTNAME3_f1': 0.6511627906976744, 'eval_PASS_f1': 0.9167945962542217, 'eval_PASSPORT_f1': 0.9394683026584867, 'eval_POSTCODE_f1': 0.9807291666666667, 'eval_SECADDRESS_f1': 0.9779069767441859, 'eval_SEX_f1': 0.9750061865874784, 'eval_SOCIALNUMBER_f1': 0.9453476983349658, 'eval_STATE_f1': 0.9803433689972629, 'eval_STREET_f1': 0.9727485076563716, 'eval_TEL_f1': 0.9676938369781312, 'eval_

 84%|████████▍ | 39500/46750 [3:34:51<34:34,  3.50it/s]   

{'loss': 0.0002, 'grad_norm': 0.018977703526616096, 'learning_rate': 1.3248247828228244e-06, 'epoch': 42.25}


 86%|████████▌ | 40000/46750 [3:37:17<32:23,  3.47it/s]

{'loss': 0.0002, 'grad_norm': 0.07507757097482681, 'learning_rate': 1.151865638502615e-06, 'epoch': 42.78}


                                                       
 86%|████████▌ | 40205/46750 [3:38:49<27:09,  4.02it/s]

{'eval_loss': 0.07124247401952744, 'eval_BOD_f1': 0.9603789836347977, 'eval_BUILDING_f1': 0.9816420193778684, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9647231749142576, 'eval_COUNTRY_f1': 0.9735182849936947, 'eval_DATE_f1': 0.9147512365434972, 'eval_DRIVERLICENSE_f1': 0.9417114510128152, 'eval_EMAIL_f1': 0.9842295268858067, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8520481927710842, 'eval_GIVENNAME2_f1': 0.7296058661778185, 'eval_IDCARD_f1': 0.9336850037678975, 'eval_IP_f1': 0.9865926953305595, 'eval_LASTNAME1_f1': 0.8008298755186721, 'eval_LASTNAME2_f1': 0.6867749419953596, 'eval_LASTNAME3_f1': 0.6648199445983378, 'eval_PASS_f1': 0.9171779141104295, 'eval_PASSPORT_f1': 0.9399917796958488, 'eval_POSTCODE_f1': 0.9825566258786774, 'eval_SECADDRESS_f1': 0.9767441860465116, 'eval_SEX_f1': 0.974764967837704, 'eval_SOCIALNUMBER_f1': 0.9450078003120124, 'eval_STATE_f1': 0.9793891234169356, 'eval_STREET_f1': 0.9717396940627431, 'eval_TEL_f1': 0.9676455948232952, 'eval

 87%|████████▋ | 40500/46750 [3:40:16<29:15,  3.56it/s]   

{'loss': 0.0002, 'grad_norm': 0.017236489802598953, 'learning_rate': 9.903113209758098e-07, 'epoch': 43.32}


 88%|████████▊ | 41000/46750 [3:42:42<25:20,  3.78it/s]

{'loss': 0.0002, 'grad_norm': 0.004363588057458401, 'learning_rate': 8.403700661183356e-07, 'epoch': 43.85}


                                                       
 88%|████████▊ | 41140/46750 [3:43:56<22:35,  4.14it/s]

{'eval_loss': 0.07163746654987335, 'eval_BOD_f1': 0.9589335626746937, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9666666666666667, 'eval_COUNTRY_f1': 0.9728877679697352, 'eval_DATE_f1': 0.9131701631701632, 'eval_DRIVERLICENSE_f1': 0.9404272801972063, 'eval_EMAIL_f1': 0.9848082035700721, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8483816013628619, 'eval_GIVENNAME2_f1': 0.7298578199052133, 'eval_IDCARD_f1': 0.9338137682533663, 'eval_IP_f1': 0.9870550161812297, 'eval_LASTNAME1_f1': 0.7989320188950503, 'eval_LASTNAME2_f1': 0.6836581709145426, 'eval_LASTNAME3_f1': 0.657608695652174, 'eval_PASS_f1': 0.9185049019607843, 'eval_PASSPORT_f1': 0.9410798603982755, 'eval_POSTCODE_f1': 0.9817803227485685, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.9745238684145436, 'eval_SOCIALNUMBER_f1': 0.9456881448316138, 'eval_STATE_f1': 0.9801291604570292, 'eval_STREET_f1': 0.9719771665801765, 'eval_TEL_f1': 0.9678384442782348, 'eval

 89%|████████▉ | 41500/46750 [3:45:42<26:25,  3.31it/s]   

{'loss': 0.0001, 'grad_norm': 0.03523299843072891, 'learning_rate': 7.022351411174866e-07, 'epoch': 44.39}


 90%|████████▉ | 42000/46750 [3:48:07<23:06,  3.43it/s]

{'loss': 0.0001, 'grad_norm': 0.0019159680232405663, 'learning_rate': 5.760845953596527e-07, 'epoch': 44.92}


                                                       
 90%|█████████ | 42075/46750 [3:49:03<20:15,  3.85it/s]

{'eval_loss': 0.07167435437440872, 'eval_BOD_f1': 0.9607927617406291, 'eval_BUILDING_f1': 0.9813346970084378, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9654496446949278, 'eval_COUNTRY_f1': 0.9738582677165355, 'eval_DATE_f1': 0.9159150421879546, 'eval_DRIVERLICENSE_f1': 0.9412492269635128, 'eval_EMAIL_f1': 0.9848024316109423, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8492753623188406, 'eval_GIVENNAME2_f1': 0.7250698974836906, 'eval_IDCARD_f1': 0.9339390497823206, 'eval_IP_f1': 0.9863646868500116, 'eval_LASTNAME1_f1': 0.8007456503728253, 'eval_LASTNAME2_f1': 0.6884996191926885, 'eval_LASTNAME3_f1': 0.6592178770949721, 'eval_PASS_f1': 0.9188361408882083, 'eval_PASSPORT_f1': 0.9408150727012083, 'eval_POSTCODE_f1': 0.9822916666666667, 'eval_SECADDRESS_f1': 0.9778812572759022, 'eval_SEX_f1': 0.974764967837704, 'eval_SOCIALNUMBER_f1': 0.9459933710274907, 'eval_STATE_f1': 0.9798657718120805, 'eval_STREET_f1': 0.9717250324254215, 'eval_TEL_f1': 0.9666666666666668, 'eval

 91%|█████████ | 42500/46750 [3:51:08<22:55,  3.09it/s]   

{'loss': 0.0001, 'grad_norm': 0.004615441896021366, 'learning_rate': 4.620810309330803e-07, 'epoch': 45.45}


 92%|█████████▏| 43000/46750 [3:53:33<18:51,  3.31it/s]

{'loss': 0.0001, 'grad_norm': 0.0005737920291721821, 'learning_rate': 3.603713930414676e-07, 'epoch': 45.99}


                                                       
 92%|█████████▏| 43010/46750 [3:54:09<16:00,  3.89it/s]

{'eval_loss': 0.07144143432378769, 'eval_BOD_f1': 0.9607758620689655, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9659230203481246, 'eval_COUNTRY_f1': 0.9741488020176544, 'eval_DATE_f1': 0.9156486329261198, 'eval_DRIVERLICENSE_f1': 0.9411521784018171, 'eval_EMAIL_f1': 0.9859369061193461, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8503974945796193, 'eval_GIVENNAME2_f1': 0.7335190343546889, 'eval_IDCARD_f1': 0.9338374291115312, 'eval_IP_f1': 0.9863646868500116, 'eval_LASTNAME1_f1': 0.799585062240664, 'eval_LASTNAME2_f1': 0.687311178247734, 'eval_LASTNAME3_f1': 0.6666666666666666, 'eval_PASS_f1': 0.920012258657677, 'eval_PASSPORT_f1': 0.9410560785918953, 'eval_POSTCODE_f1': 0.9823008849557522, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.974764967837704, 'eval_SOCIALNUMBER_f1': 0.9463414634146342, 'eval_STATE_f1': 0.9803629132488193, 'eval_STREET_f1': 0.9724818276220146, 'eval_TEL_f1': 0.966940094456873, 'eval_TIM

 93%|█████████▎| 43500/46750 [3:56:33<16:22,  3.31it/s]   

{'loss': 0.0001, 'grad_norm': 0.054611772298812866, 'learning_rate': 2.7108678059855064e-07, 'epoch': 46.52}


                                                       
 94%|█████████▍| 43945/46750 [3:59:16<11:54,  3.93it/s]

{'eval_loss': 0.07177942246198654, 'eval_BOD_f1': 0.9607927617406291, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9659397206567018, 'eval_COUNTRY_f1': 0.9738582677165355, 'eval_DATE_f1': 0.9169338385310405, 'eval_DRIVERLICENSE_f1': 0.9403046521202141, 'eval_EMAIL_f1': 0.9859369061193461, 'eval_GEOCOORD_f1': 0.9813953488372094, 'eval_GIVENNAME1_f1': 0.8502763758711849, 'eval_GIVENNAME2_f1': 0.7313019390581718, 'eval_IDCARD_f1': 0.9345226798253938, 'eval_IP_f1': 0.9863646868500116, 'eval_LASTNAME1_f1': 0.8009950248756219, 'eval_LASTNAME2_f1': 0.6922486569455104, 'eval_LASTNAME3_f1': 0.6630434782608696, 'eval_PASS_f1': 0.9190184049079755, 'eval_PASSPORT_f1': 0.9413207932938049, 'eval_POSTCODE_f1': 0.9815152304087478, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.9752475247524752, 'eval_SOCIALNUMBER_f1': 0.9462785700332096, 'eval_STATE_f1': 0.9796222664015904, 'eval_STREET_f1': 0.972229431611731, 'eval_TEL_f1': 0.9674372358936117, 'eval

 94%|█████████▍| 44000/46750 [3:59:33<13:48,  3.32it/s]  

{'loss': 0.0001, 'grad_norm': 0.0016892210114747286, 'learning_rate': 1.9434227724779987e-07, 'epoch': 47.06}


 95%|█████████▌| 44500/46750 [4:01:59<10:16,  3.65it/s]

{'loss': 0.0001, 'grad_norm': 0.056727029383182526, 'learning_rate': 1.3023680302504338e-07, 'epoch': 47.59}


                                                       
 96%|█████████▌| 44880/46750 [4:04:23<07:47,  4.00it/s]

{'eval_loss': 0.07161174714565277, 'eval_BOD_f1': 0.9608096468561584, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9651960784313726, 'eval_COUNTRY_f1': 0.9744721084147495, 'eval_DATE_f1': 0.9176401869158879, 'eval_DRIVERLICENSE_f1': 0.9403046521202141, 'eval_EMAIL_f1': 0.9857495724871745, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8506353392471829, 'eval_GIVENNAME2_f1': 0.7304832713754646, 'eval_IDCARD_f1': 0.9349516404323914, 'eval_IP_f1': 0.9868208092485549, 'eval_LASTNAME1_f1': 0.8005804311774462, 'eval_LASTNAME2_f1': 0.6855828220858895, 'eval_LASTNAME3_f1': 0.6666666666666666, 'eval_PASS_f1': 0.9208588957055214, 'eval_PASSPORT_f1': 0.9408393039918118, 'eval_POSTCODE_f1': 0.9815152304087478, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.9750061865874784, 'eval_SOCIALNUMBER_f1': 0.9470392808286106, 'eval_STATE_f1': 0.9801093983092989, 'eval_STREET_f1': 0.9709392838609238, 'eval_TEL_f1': 0.9671641791044777, 'eva

 96%|█████████▋| 45000/46750 [4:04:58<08:06,  3.60it/s]  

{'loss': 0.0002, 'grad_norm': 0.031119778752326965, 'learning_rate': 7.885298685522235e-08, 'epoch': 48.13}


 97%|█████████▋| 45500/46750 [4:07:25<06:42,  3.11it/s]

{'loss': 0.0001, 'grad_norm': 0.0029458084609359503, 'learning_rate': 4.025706004760932e-08, 'epoch': 48.66}


                                                       
 98%|█████████▊| 45815/46750 [4:09:29<03:35,  4.34it/s]

{'eval_loss': 0.07161623984575272, 'eval_BOD_f1': 0.9608096468561584, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9649595687331537, 'eval_COUNTRY_f1': 0.9744721084147495, 'eval_DATE_f1': 0.9173722627737226, 'eval_DRIVERLICENSE_f1': 0.9403046521202141, 'eval_EMAIL_f1': 0.9857495724871745, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8495192307692307, 'eval_GIVENNAME2_f1': 0.7303058387395738, 'eval_IDCARD_f1': 0.9349516404323914, 'eval_IP_f1': 0.9868208092485549, 'eval_LASTNAME1_f1': 0.8009950248756219, 'eval_LASTNAME2_f1': 0.6885496183206107, 'eval_LASTNAME3_f1': 0.6611111111111112, 'eval_PASS_f1': 0.920245398773006, 'eval_PASSPORT_f1': 0.9406224406224406, 'eval_POSTCODE_f1': 0.98125, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.9750061865874784, 'eval_SOCIALNUMBER_f1': 0.9466484268125855, 'eval_STATE_f1': 0.9801093983092989, 'eval_STREET_f1': 0.972229431611731, 'eval_TEL_f1': 0.9671641791044777, 'eval_TIME_f1': 0

 98%|█████████▊| 46000/46750 [4:10:25<03:55,  3.19it/s]  

{'loss': 0.0001, 'grad_norm': 0.004590354394167662, 'learning_rate': 1.449877092679075e-08, 'epoch': 49.2}


 99%|█████████▉| 46500/46750 [4:12:50<01:11,  3.49it/s]

{'loss': 0.0001, 'grad_norm': 0.01150474138557911, 'learning_rate': 1.6113207094181626e-09, 'epoch': 49.73}


                                                       
100%|██████████| 46750/46750 [4:14:37<00:00,  3.77it/s]

{'eval_loss': 0.07163183391094208, 'eval_BOD_f1': 0.9608096468561584, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9649595687331537, 'eval_COUNTRY_f1': 0.9744721084147495, 'eval_DATE_f1': 0.9173722627737226, 'eval_DRIVERLICENSE_f1': 0.9400864731315627, 'eval_EMAIL_f1': 0.9857495724871745, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8495192307692307, 'eval_GIVENNAME2_f1': 0.7303058387395738, 'eval_IDCARD_f1': 0.9345971563981043, 'eval_IP_f1': 0.9868208092485549, 'eval_LASTNAME1_f1': 0.8009950248756219, 'eval_LASTNAME2_f1': 0.6885496183206107, 'eval_LASTNAME3_f1': 0.6611111111111112, 'eval_PASS_f1': 0.920245398773006, 'eval_PASSPORT_f1': 0.9406224406224406, 'eval_POSTCODE_f1': 0.9815152304087478, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.9750061865874784, 'eval_SOCIALNUMBER_f1': 0.9464425332290852, 'eval_STATE_f1': 0.9801093983092989, 'eval_STREET_f1': 0.9719771665801765, 'eval_TEL_f1': 0.9671641791044777, 'eval

100%|██████████| 46750/46750 [4:14:38<00:00,  3.06it/s]

{'train_runtime': 15278.6034, 'train_samples_per_second': 97.875, 'train_steps_per_second': 3.06, 'train_loss': 0.032043613471768116, 'epoch': 50.0}


TrainOutput(global_step=46750, training_loss=0.032043613471768116, metrics={'train_runtime': 15278.6034, 'train_samples_per_second': 97.875, 'train_steps_per_second': 3.06, 'total_flos': 8.79207700629273e+16, 'train_loss': 0.032043613471768116, 'epoch': 50.0})

{'eval_loss': 0.07163183391094208, 'eval_BOD_f1': 0.9608096468561584, 'eval_BUILDING_f1': 0.9815856777493606, 'eval_CARDISSUER_f1': 1.0, 'eval_CITY_f1': 0.9649595687331537, 'eval_COUNTRY_f1': 0.9744721084147495, 'eval_DATE_f1': 0.9173722627737226, 'eval_DRIVERLICENSE_f1': 0.9400864731315627, 'eval_EMAIL_f1': 0.9857495724871745, 'eval_GEOCOORD_f1': 0.9836829836829837, 'eval_GIVENNAME1_f1': 0.8495192307692307, 'eval_GIVENNAME2_f1': 0.7303058387395738, 'eval_IDCARD_f1': 0.9345971563981043, 'eval_IP_f1': 0.9868208092485549, 'eval_LASTNAME1_f1': 0.8009950248756219, 'eval_LASTNAME2_f1': 0.6885496183206107, 'eval_LASTNAME3_f1': 0.6611111111111112, 'eval_PASS_f1': 0.920245398773006, 'eval_PASSPORT_f1': 0.9406224406224406, 'eval_POSTCODE_f1': 0.9815152304087478, 'eval_SECADDRESS_f1': 0.9773123909249564, 'eval_SEX_f1': 0.9750061865874784, 'eval_SOCIALNUMBER_f1': 0.9464425332290852, 'eval_STATE_f1': 0.9801093983092989, 'eval_STREET_f1': 0.9719771665801765, 'eval_TEL_f1': 0.9671641791044777, 'eval_TIME_f1': 0.9675384413194901, 'eval_TITLE_f1': 0.9590373783922173, 'eval_USERNAME_f1': 0.9522776572668112, 'eval_precision': 0.9357036693883066, 'eval_recall': 0.9483898736158392, 'eval_f1': 0.9420040613674567, 'eval_accuracy': 0.9914942665289909, 'eval_runtime': 33.1297, 'eval_samples_per_second': 239.846, 'eval_steps_per_second': 7.516, 'epoch': 50.0}
